---
title: 玉米籽粒脱水
subtitle:
date: 2024-11-16T22:24:12+08:00
draft: false
type: posts
author:
  name:
  link:
  email:
  avatar:
description:
keywords:
license:
comment: false
weight: 0
tags:
  - 文献阅读
categories:
  - 文献阅读
hiddenFromHomePage: false
hiddenFromSearch: false
summary:
resources:
  - name: featured-image
    src: featured-image.jpg
  - name: featured-image-preview
    src: featured-image-preview.jpg
toc: true
math: false
lightgallery: false
password:
message:
repost:
  enable: true
  url:

# See details front matter: https://fixit.lruihao.cn/documentation/content-management/introduction/#front-matter
---


题目：[A Zea genus-specific micropeptide controls kernel  dehydration in maize](https://www.cell.com/cell/fulltext/S0092-8674(24)01212-1)   
文章单位：华中农业大学  
通讯作者：[严建兵](https://cpst.hzau.edu.cn/info/1015/1602.htm)    

- [Abstract](#Abstract)
- [Introduction](#Introduction)
- [Result](#Result)
- [Method](#Method)
- [Fig](#Fig)

## Abstract {#Abstract}

&emsp;  Kernel dehydration rate (KDR) is a crucial production trait that affects mechanized harvesting and kernel  quality in maize; however, the underlying mechanisms remain unclear. Here, we identified a quantitative trait  locus (QTL), qKDR1, as <font color="darkred" ><b>a non-coding sequence that regulates the expression of qKDR1 REGULATED PEPTIDE GENE (RPG)</b></font>. RPG encodes a 31 amino acid micropeptide, microRPG1, which controls KDR by precisely  modulating the expression of two genes, ZmETHYLENE-INSENSITIVE3-like 1 and 3, in the ethylene signaling  pathway in the kernels after filling. microRPG1 is a Zea genus-specific micropeptide and originated de novo  from a non-coding sequence. Knockouts of microRPG1 result in faster KDR in maize. By contrast, overexpression or exogenous application of the micropeptide shows the opposite effect both in maize and Arabidopsis. Our findings reveal the molecular mechanism of microRPG1 in kernel dehydration and provide an  important tool for future crop breeding.

- Kernel 籽粒; dehydration 脱水; 
- [从头起源](https://future.pku.edu.cn/xwyjz/xw/08f334eb3c344990a80adcc99c70feba.htm#:~:text=%E9%95%BF%E6%9C%9F%E4%BB%A5%E6%9D%A5%EF%BC%8C%E4%BA%BA%E4%BB%AC%E8%AE%A4%E4%B8%BA%E6%96%B0%E5%9F%BA%E5%9B%A0%E4%B8%BB%E8%A6%81%E9%80%9A%E8%BF%87%E5%9F%BA%E5%9B%A0%E5%A4%8D%E5%88%B6%EF%BC%88gene%20duplication%EF%BC%89%E4%BA%A7%E7%94%9F%EF%BC%8C%E8%80%8C%E8%BF%91%E5%B9%B4%E7%9A%84%E7%A0%94%E7%A9%B6%E5%8F%91%E7%8E%B0%E4%BB%8E%E5%A4%B4%E8%B5%B7%E6%BA%90%EF%BC%88,de%20novo%20origination%EF%BC%89%E5%90%8C%E6%A0%B7%E5%9C%A8%E6%96%B0%E5%9F%BA%E5%9B%A0%E4%BA%A7%E7%94%9F%E4%B8%AD%E5%8F%91%E6%8C%A5%E9%87%8D%E8%A6%81%E4%BD%9C%E7%94%A8%EF%BC%9A%E5%9C%A8%E8%BF%99%E4%B8%80%E6%A8%A1%E5%9E%8B%E4%B8%8B%EF%BC%8C%E6%96%B0%E5%9F%BA%E5%9B%A0%E4%BB%8E%E7%A5%96%E5%85%88%E9%9D%9E%E7%BC%96%E7%A0%81%E5%8C%BA%E5%9F%9F%E6%A8%AA%E7%A9%BA%E5%87%BA%E4%B8%96%EF%BC%8C%E5%BD%A2%E6%88%90%E5%89%8D%E6%89%80%E6%9C%AA%E6%9C%89%E7%9A%84%E7%BC%96%E7%A0%81%E8%9B%8B%E7%99%BD%E3%80%82)

## Introduction {#Introduction}

&emsp; Maize is one of the most important crops in the world, with an annual global production of 1.21 billion tons in 2021 [FAO](https://www.fao.org/). Mechanized harvesting of maize kernels is an  effective way to solve the continuous increase of labor costs  and improve production efficiency. However, mechanized harvesting has not yet been achieved in China and many other  countries, and the mechanized harvesting area of maize kernel  is less than 5% in China, partly due to a lack of suitable maize  varieties.<sup>1–3</sup> Kernel dehydration rate (KDR) is a key determinant of moisture content at harvest that affects mechanized  harvesting and kernel quality.<sup>4–6</sup> A high moisture content at  harvest seriously limits harvesting efficiency and increases  the cost of drying and storage.<sup>7,8</sup> <font color="darkred"><b>This represents a considerable problem as most maize varieties in China have high grain  water content at harvest, ranging from 30% to 40%, but the  moisture content suitable for mechanized harvesting is from 15% to 25%, seriously limiting the widespread application of  mechanical harvesting</b></font>.<sup>1,9–12</sup> Farmers usually delay harvest  to reduce moisture content, which leads to a delay in the  planting time of the next crop and greatly affects the annual  yield of the crops, especially in the Huang-Huai-Hai region  of China with a Wheat-Maize Rotation System.<sup>1,4,7</sup> High moisture content can also lead to kernel breakage during mechanical harvesting,<sup>6,13,14</sup> and can cause kernel mold, ear rot, and ear sprouting under warm and humid environmental conditions.<sup>10,15–17</sup> Thus, increasing KDR and reducing kernel  moisture content at harvest is essential and has become a major goal of modern maize breeding.<sup>18,19</sup>

&emsp; Many studies have identified quantitative trait loci (QTLs) controlling KDR in maize over the past two decades.<sup>1,4,6,10,13,18,20–23</sup>  However, very few QTLs have been cloned,<sup>21</sup> rendering the molecular mechanism of maize KDR unsolved. Furthermore, the  genes controlling KDR are also rarely reported in plants,<sup>24</sup> and  the underlying mechanisms are largely unknown.

&emsp; Recent advances have shown that some so-called long noncoding RNAs (lncRNAs) have small open reading frames  (sORFs) less than 100 amino acids in length.<sup>25–27</sup> These sORFs can be translated into small proteins, often referred to as micropeptides.<sup>28–30</sup> Many functional peptides translated from  lncRNAs have been studied in humans, leading to molecular insights into their functions.<sup>30–34</sup> While a few functional micropeptides have been identified in plants,<sup>35–40</sup> relatively little is known  about their roles.

&emsp; Furthermore, the origin of new genes can generate and maintain remarkable phenotypic diversity.<sup>41–43</sup> These new genes can originate de novo from non-coding sequences and  affect adaptation, morphological diversity, and speciation of organisms.<sup>43–47</sup> However, new genes in plants that are both novel and phenotypically impactful remain largely uncharacterized.<sup>48</sup>

&emsp; In this study, we fine-mapped and cloned a QTL for KDR in  maize, Kernel Dehydration Rate1 (qKDR1), to a non-coding intergenic DNA sequence. qKDR1 regulates the expression of a 31 aa  micropeptide encoding gene, qKDR1 Regulated Peptide Gene
(RPG), located 10 kilobases upstream of qKDR1. Two MYB-related transcription factors, ZmMYBST1 and ZmMYBR43,  bind to the qKDR1 region to repress RPG expression. microRPG1 (micropeptide of RPG ORF1) is derived from a sORF  generated by a single nucleotide mutation at the base of the  genus Zea that created a start codon. The microRPG1 micropeptide controls KDR by regulating the expression of genes in  the ethylene signaling pathway ZmETHYLENE-INSENSITIVE3-like 1 and 3 (ZmEIL1 and 3). Modulation of microRPG1 expression or exogenous application of the peptide in the distantly  related dicot model Arabidopsis impacts KDR. Our findings  shed light on the molecular mechanism underlying KDR and provide useful insights for future crop breeding, underscore the significant role of micropeptides, and present an excellent model for de novo gene evolution.
- distantly 远离地，遥远地；疏远地，冷淡地；模糊地; dicot model 双子叶模式植物; underscore 强调

## Result {#Result}

### qKDR1 is the causal locus for KDR

&emsp; We mapped four QTLs for KDR in a maize recombinant inbred  line (RIL) population developed from a cross between the maize  inbred lines K22 and DAN340<sup>49,50</sup> (Figure 1A). The QTL with the largest effect, qKDR1 on the long arm of chromosome 1 and ex-  plaining 9.8% of the phenotypic effect, was selected for posi-  tional cloning. To examine the allelic effects, we developed a  pair of qKDR1 near isogenic lines (NILs) (NILDAN340 and NILK22)  (Figure 1B) from a heterogeneous inbred family library.54  Compared with NILDAN340, NILK22 had a more rapid KDR with  up to 1.8% reduction in moisture content at harvest, but no  obvious difference in other agronomic traits (Figure 1B;  Table S1).
&emsp; The QTL was further narrowed down to a 1,417 base pair (bp) intergenic non-coding region according to the maize B73 refer-  ence sequence version 4.0 (Figure 1C). NILDAN340 had a 6,181-  bp insertion in this region compared with NILK22 (Figure 1C;  Table S2). To confirm the function of qKDR1, we deleted the  target sequence using CRISPR-Cas9 in two different maize  inbred lines, Zheng58 (similar to NILK22) and B104 (similar to  NILDAN340) (Figure  1D).  Both  knockout  lines  had  slower KDR
compared with wild-type siblings (Figures 1E and 1F), with few  effects on other agronomic traits in two different environments  (Table S3). We next surveyed the natural variation of the 6,181-  bp QTL region insertion in a diverse maize inbred line popula-  tion,55 and identified two additional insertions or deletions (in-  dels) in this region by polymerase chain reaction (PCR). Finally,  five different haplotypes were identified (Figure 1G), and four of  them, Hap2–Hap5 (with indel), had a lower KDR compared  with Hap1 (Figure 1H). The collective data all show statistically  significant differences based on different methods (Figure 1I).  These results indicate that the 1,417-bp sequence of qKDR1 is  a key determinant for KDR variation, as knocking out this  sequence resulted in a reduced KDR in maize.

### qKDR1 acts as a silencer

&emsp; An intriguing question is, as a non-coding sequence, how does  the 1,417-bp sequence function. To answer this question, we  performed transient transcriptional activity assays in maize pro-  toplasts. The qKDR1 sequences from the two NILs, qKDR12D  (a qKDR1 fragment of NILDAN340) and qKDR12K (a qKDR1 frag-  ment of NILK22), were independently fused downstream of the  b-glucuronidase (GUS) gene driven by a minimal Ubiquitin pro-  moter (mpUbi) (named pUbi2D and pUbi2K) and using the firefly  luciferase (LUC) gene as an internal control (Figures 2A and 2B).  Compared with GUS activity of the mpUbi vector (pUbiVector),  the activities of pUbi2D and pUbi2K were both significantly  reduced. Moreover, the GUS activity of pUbi2K was lower than  pUbi2D, demonstrating that GUS expression was reduced by  the presence of both qKDR1 alleles, and the inhibitory effect of  qKDR12K was greater than that of qKDR12D (Figures 2A–2C).  This suggests that the different qKDR1 alleles act to repress  target gene expression. We next fused a series of qKDR1 frag-  ments with different lengths downstream of the GUS gene  (named pUbi3D and pUbi3K-pUbi5K) (Figures 2A and 2B). Tran-  sient transcriptional activity assays indicated that all of them dis-  played a similar repressive activity to pUbi2K (Figures 2A–2C),  but pUbi2D led to higher GUS activity. These results suggest  that a 369-bp sequence of qKDR1 is the core silencer sequence  and that the 6,181-bp insertion in NILDAN340 reduces the  silencing activity.

### RPG is the qKDR1 target gene

&emsp; To identify the gene(s) regulated by qKDR1, we performed RNA  sequencing (RNA-seq) on four  NILs harboring different  qKDR1 genotypes (Figure 2D). We rationalized that genes that were  differentially expressed according to the genotypes of qKDR1  would be candidate target genes (Figures 2E and 2F). We iden-  tified 17 genes, of which three were upregulated and 14 were  downregulated (Figures 2E and 2F; Table S4). Among these  genes, only one, which we named RPG, was close to qKDR1, being located 10-kb upstream (Figure 1C). Consistent with the KDR phenotype, RPG was highly expressed in maize kernels af-  ter 30 days after pollination (DAP) (Figures S1D, S4A, and S4B).  Real-time quantitative PCRs (qPCRs) revealed that RPG expres-  sion was lower in NILK22 than in NILDAN340 in the late stage of  maize kernel maturation (Figure 2G). In the diverse inbred line  population, we also found that the RPG expression in Hap2  and Hap5 (with indels in qKDR1) was significantly higher than
that in Hap1 (without indel in qKDR1) (Figure 2H). As expected, from H3K4me3 and RNA polymerase II (RNAPII) occupancy56  RPG expression was also higher in both of the qKDR1 knockout were also detected between RPG and the qKDR1 locus  lines (Figure 2I). Furthermore, chromatin interactions inferred (Figure 2J).

&emsp; To further confirm that RPG is regulated by qKDR1, we  fused the 883-bp RPG promoter upstream of the firefly LUC  gene (named pRPGVector), and the Renilla luciferase (REN)  gene was used as an internal control. The qKDR1 fragments  of different lengths from NILDAN340 or NILK22 were also  fused downstream of the LUC gene (named pRPG3D and  pRPG3K-pRPG5K) (Figures S1A and S1B). Compared with  the RPG promoter-driven LUC activity, all of them displayed  lower LUC activity (Figures S1A–S1C). This result is consistent  with transient transcriptional activity assays of mpUbi-driven  GUS activity (Figures 2A–2C). These results thus collectively  suggest that qKDR1 functions as a repressor of RPG expres-  sion, and the 369-bp region of qKDR1 is sufficient for silencing  activity.

&emsp; Published chromatin immunoprecipitation sequencing (ChIP-seq) data57 showed that 43 transcription factors can bind to  the qKDR1 region, two of which are annotated as transcriptional  repressors (Table S5, Zm00001d000184, MYB-related-tran-  scription factor 31, named ZmMYBST1 and Zm00001d029875,  MYB-related-transcription factor 43, named ZmMYBR43),  consistent with the role of qKDR1 acting as a silencer.  ZmMYBST1 and ZmMYBR43 were expressed in a similar pattern  as RPG with sequence similarity of 96% (Figure S1D),58,59 sug-  gesting that they may affect RPG expression. mRNA in situ hy-  bridization was performed to confirm the expression patterns  of ZmMYBST1, ZmMYBR43, and RPG, and the results showed  that they were all expressed in the embryo, aleurone layer, and  placento-chalazal region (Figure S1E). To evaluate the effects  of ZmMYBST1 and ZmMYBR43 on the qKDR1-regulated  expression of RPG, we performed transient transcriptional activ-  ity assays, in which the coding sequence of ZmMYBST1 and  ZmMYBR43 driven by the CaMV 35S promoter were used as ef-  fectors, and pRPG3D and pRPG3K were used as the reporters,  respectively (Figure S1F). Compared with the control, co-  expression with ZmMYBST1 or ZmMYBR43 resulted in strongly  repressed LUC activities of the two reporters (Figure S1G). Addi-  tionally, double mutants of ZmMYBST1 and ZmMYBR43 gener-  ated by CRISPR-Cas9 significantly decelerated KDR compared  with wild-type siblings (Figure S1H). These results suggest that  ZmMYBST1 and ZmMYBR43 target the qKDR1 region to  repress RPG expression in vivo.

### RPG encodes a functional micropeptide

&emsp; RPG  was not  annotated as a gene in  maize  (B73 version 4.0). However, high-depth RNA-seq reads mapped to and were pre-  sumably transcribed by the RPG region (Figure 3A). We thus per-  formed rapid-amplification of cDNA ends (RACE), including both  50-RACE and 30-RACE, and found two full-length RPG tran-  scripts, named T01 and T02, with lengths of 2,013 and  1,723 bp in NILK22 (Figure 3B; Table S2). Next, to identify  possible protein(s) encoded by RPG, we screened putative  ORFs of the two transcripts using ORF finder (https://www.  ncbi.nlm.nih.gov/orffinder) and identified twelve putative ORFs  (Figure 3C), with the largest encoding a micropeptide of 58  amino acids. Small RNA-seq (sRNA-seq) revealed that RPG  likely does not function by producing sRNAs (Figure 3D). Howev-  er, ribosome profiling sequencing (Ribo-seq) revealed that RPG  mRNA was ribosome bound in the ORF1, ORF2, and ORF3 regions, suggesting that it encodes one or more micropeptides  (Figure 3E).

&emsp; To assess whether RPG functions by encoding micropeptides  rather than a lncRNA, we overexpressed the full-length RPG or a  mutant RPG with all 12 ATG start codons mutated to CTG (Fig-  ure S2E). Overexpression of wild-type RPG resulted in a slower  KDR (Figures S2A–S2D), whereas the non-translatable mutant  RPG did not alter the KDR (Figures S2E–S2H). These results  imply that RPG functions by encoding micropeptides rather  than as a lncRNA.

&emsp; We found two natural deletions in the RPG region in a diverse  maize inbred line population.55 These removed most of the ORF2
and ORF3 sequences, and the different haplotypes had no  obvious KDR differences (Figures S2I–S2L). We also knocked  out ORF2 and ORF3 by a 40-bp deletion induced by CRISPR-  Cas9 (Figure S2K), and these lines had similar KDR compared  with wild-type siblings (Figures S2M and S2N). By contrast, three  frameshift mutations in ORF1 with 1-, 2-, or 4-bp deletions  (Figures 3F and 3G) all led to a faster KDR (Figures 3H–3K).  These findings indicate that ORF1 encodes the functional RPG  micropeptide. We next evaluated the effect of ORF1 on the  kernel moisture content across different environments. At har-  vest, the kernel moisture content of ORF1 knockout lines  decreased between 1.8% and 17.0% (average of 7.1%) in  different environments compared with wild type (Figures 3L–  3O). Moreover, the microRPG1 knockout lines had no obvious  effects on other agronomic traits in three different environments  (Table S6). These results suggest that editing of RPG ORF1 has  potential for breeding maize with higher KDRs.

### ORF1 encodes a micropeptide
To further investigate the influence of ORF1 on the KDR pheno-  type, we expressed ORF1 and ORF1m (in which the ORF1 start  codon was mutated to CTG) fused with FLAG tags under the  control of the Ubiquitin promoter, in maize inbred line B104 (Fig-  ure 4A). The wild-type ORF1 construct, which expresses the  ORF1 micropeptide, significantly decelerated KDR (Figures 4B,  4C, S3A, and S3B). By contrast, expression of the ORF1m  construct did not alter the KDR (Figures 4D, 4E, S3C, and  S3D). These results provide additional evidence that the ORF1  micropeptide product of the RPG gene affects maize KDR.
To further confirm that the ORF1 was translated in maize, we  fused a GFPm ORF, lacking a start codon to the C terminus of  ORF1, and driven by the 883-bp RPG promoter in protoplasts  (Figure 4F). Indeed, we observed fluorescence of the ORF1-  GFPm fusion protein in transformed cells (Figure 4G). However,  mutation of the ORF1 start codon from ATG to ATT abolished the  expression of the ORF1-GFPm fusion protein (Figure 4G). We  used anti-GFP antibodies in western blotting to confirm expres-  sion of the ORF1-GFPm fusion protein, and expression was ab-  sent when the ORF1 ATG was mutated (Figure 4H). Similar re-  sults were obtained when we used the CaMV 35S promoter in  place of the RPG promoter (Figures S3E and S3F).
We next asked if we could detect the endogenously produced
RPG ORF1 micropeptide. We probed western blots of total plant  protein and different concentrations of the synthetic ORF1 mi-  cropeptide (ORF1p) using an anti-ORF1 antibody. Indeed, we  detected the ORF1 micropeptide of the expected size in protein extracts from maize kernels (Figure 4I). The endogenous ORF1  micropeptide was also verified by immunoprecipitation (IP)  (Figure S3G) and identification by mass spectrometry (MS)  (Figures 4J and S3H–S3J). These results indicate that the ORF1 micropeptide is produced endogenously in maize kernels.  Thus, we named this 31 amino acid residue micropeptide as microRPG1 (micropeptide of RPG ORF1) (Figure 4J). microRPG1  appears not related to any known peptides in public databases.

### microRPG1 may control kernel dehydration through  regulation of ethylene signaling

&emsp; To ask how microRPG1 functions, we performed RNA-seq us-  ing  microRPG1 knockout and overexpression lines compared
with wild type as well as Arabidopsis treated with microRPG1.  384 genes were significantly downregulated in the microRPG1  knockout line and upregulated in microRPG1 overexpression  line  and exogenous micropeptide application, and 121  genes showed an opposite trend (Figures 5A and 5B). RPG is highly  expressed in the late stage of maize kernel development (after
30 DAP) when the dry matter has been completely filled  (Figures  S1D, S4A, and  S4B), implying  that  microRPG1 may
function only during specific developmental periods. Since  RPG is specifically expressed in the late stage of kernel matu-  ration, we screened for differentially expressed genes with  similar  expression  patterns  to  RPG  by  Z  score  (R2).  As a result, 11 genes (Figure S4C) had similar expression pattern to  RPG in seed, endosperm, and embryo (Figure 5C), and these  genes are more likely to be regulated by microRPG1 and play  a similar role in the late stage of maize kernel development.  Among them, Zm00001d047563 an ortholog of Ethylene-  Insensitive3/Ethylene-Insensitive3-like 1 (EIN3/EIL1), a key  component in ethylene signaling in Arabidopsis and rice  (Figure 5D),60–65 had the highest expression (Figure 5C).  Zm00001d047563 has four paralogs in maize; however, only  Zm00001d047563   (named   ZmEIL1)   and Zm00001d028974
(named ZmEIL3) were expressed in the kernel and had similar  expression pattern to RPG (Figures S1E and S4D). To  confirm
the expression pattern of microRPG1 micropeptide, we per-  formed immunohistochemistry. The results showed that it  was expressed in the embryo, aleurone layer, and placento-  chalazal region (Figure S4E), consistent with the mRNA  expression of RPG (Figure S1E). qPCR confirmed that  ZmEIL1 and ZmEIL3 were upregulated in the microRPG1  knockout and downregulated in the microRPG1 overexpres-  sion lines (Figures 5E and 5F). This suggests that microRPG1  expression in the late stage of maize kernel development  inhibits ethylene signaling, resulting in slower dehydration.  Similarly, the Arabidopsis paralogs EIN3 and EIL1 were down-  regulated following exogenous application of microRPG1  (Figures 5G and 5H). These results indicate that microRPG1  represses the expression of EIN3-related genes in both maize  and Arabidopsis.
To investigate the influence of ZmEIL1 and ZmEIL3 on the KDR  phenotype, we next  generated two knockout  lines for  ZmEIL1
and three for ZmEIL3 by CRISPR-Cas9 (Figures S4F and S4G).  All of the knockout lines significantly decelerated KDR compared  with wild-type sibling plants (Figures 5I and 5J). We also investi-
gated the effects of ZmEIL1 or ZmEIL3 knockout on kernel qual-  ity traits and seed germination rate and found that knockout of  EIL genes affected germination rate significantly and kernel qual-
ity slightly (Figures S5A–S5L). These results suggest that EIL  genes play important roles in plant development and that the  fine-tuning of their expression by microRPG1 can be  beneficial
for agricultural production.
Considering that EIN3/EIL1 are key components in ethylene  signaling, we also applied exogenous ethylene to different  maize    lines   and   found    that  this    could   accelerate KDR
(Figures 5K–5M). Furthermore, we tested five marker genes  for ethylene signaling62,66–72 in microRPG1 knockout and over-  expression lines and following exogenous micropeptide appli-  cation. We found that these marker genes were downregulated  when microRPG1 was overexpressed or exogenously applied  and upregulated when microRPG1 was knocked out, support-  ing the idea that microRPG1 regulates ethylene signaling  (Figures S5M–S5O). Our results indicate that ethylene signaling  can affect KDR, suggesting a possible downstream mechanism for microRPG1 in controlling this trait. This finding is consistent  with the well-known role of ethylene in stimulating fruit  ripening.<sup>71–73</sup>

### Origin of microRPG1 micropeptide
To investigate the origins of the microRPG1 micropeptide, we  analyzed its syntenic regions in the Poaceae. We found homolo-  gous sequences in the genus Zea and Tripsacum but absent  from other members of the Poaceae. Sequence alignments indi-  cated that the 96-bp microRPG1 sequence was highly  conserved in all maize and teosinte genomes (Table S2).  Although similar sequences were present in Tripsacum dacty-  loides, they lacked a start codon, suggesting they are not trans-  lated. Phylogenetic analysis showed that a nucleotide (ACG to  ATG) mutation led to the evolution of the ORF1 micropeptide  (Figures 6, S6A, and S6B). These observations, together with  the above finding that no known peptide associations exist in  public databases, suggest that the microRPG1 micropeptide  originated de novo from a non-coding sequence after the recent split between the genus Zea and Tripsacum 0.65 million years ago (mya) (Figure 6).45,74–78

### microRPG1 functions similarly in Arabidopsis

&emsp; Given our demonstration that the microRPG1 micropeptide is  able to control the ancient seed dehydration pathway in maize, we wondered whether it could exert similar effects in distantly  related plant species. We therefore investigated whether exogenous application of synthetic ORF1p would function in Arabi-  dopsis (Col-0). We applied different ORF1p concentrations  ranging from 0.01 to 2 mM. Remarkably, silique ripening was  significantly delayed at peptide concentrations of 1 and 2 mM  (Figures S7A and S7B). A 2 mM concentration of ORF1p was  selected for additional studies that again delayed silique  ripening compared with plants treated with a control peptide  (scORF1p), with the same amino acid composition but a  scrambled sequence (Figures 7A–7E). Application of ORF1p  significantly increased the moisture content of Arabidopsis  seeds, but there was no significant difference in flowering  time (Figures 7F and 7G). This result is similar to microRPG1  overexpression in maize, which leads to a slower KDR. To  confirm this result, we generated three microRPG1 overexpres-  sion lines in Arabidopsis, and all of them significantly delayed  silique ripening (Figures S7C and S7D). Our results indicate  that microRPG1 micropeptide functions similarly in the distantly  related Arabidopsis species as in maize.
The peptide treatment experiment suggests that ORF1p can  enter cells, presumably by crossing the plasma membrane,  echoing the behavior of some reported peptides.79,80 To test  this idea, we labeled ORF1p with fluorescein (FAM, carboxy-  fluorescein) (FAM-ORF1p) and incubated it with Arabidopsis  seedlings. Interestingly, the fluorescein-labeled ORF1p rapidly penetrated into the root ( 0.5 h) but took longer to penetrate other parts of the seedling (Figures S7E and S7F). At 24h after  application, FAM-ORF1p was detected in most parts of the  root and began to appear in the cotyledons (Figures 7H and  S7E). FAM-ORF1p was detected in leaf veins (Figures 7H  and S7G), suggesting that it is transported to the shoot  tissues. To address whether ORF1p acts intracellularly in  Arabidopsis rather than through cell surface receptors, we  observed the subcellular localization of FAM-ORF1p. We  found that FAM-ORF1p could enter root cells and localized  in the nucleus, cytoplasm, and plasma membrane  (Figures S7H and S7I), suggesting it may function through  different mechanisms. These results were confirmed by incu-  bation of FAM-ORF1p with maize protoplasts (Figure S7J).  We also fused ORF1 with GFP and transformed it into maize  protoplasts, and observed fluorescence in the nucleus, cyto-  plasm, and plasma membrane (Figure S7K). Taken together,  these data suggest that the ORF1p micropeptide can enter  cells and plays a similar role in Arabidopsis and maize,  implying conserved function across plant species.















## Method {#Method}
## Fig {#Fig} 

