---
title: 玉米多组学网络
subtitle:
date: 2024-09-05T08:58:46+08:00
draft: false
type: posts
author:
  name:
  link:
  email:
  avatar:
description:
keywords:
license:
comment: false
weight: 0
tags:
  - 文献阅读
categories:
  - 文献阅读
hiddenFromHomePage: false
hiddenFromSearch: false
summary:
resources:
  - name: featured-image
    src: featured-image.jpg
  - name: featured-image-preview
    src: featured-image-preview.jpg
toc: true
math: false
lightgallery: false
password:
message:
repost:
  enable: true
  url:

# See details front matter: https://fixit.lruihao.cn/documentation/content-management/introduction/#front-matter
---

<!--more-->
题目：[A multi-omics integrative network map of maize](https://www.nature.com/articles/s41588-022-01262-1)  
文章单位：华中农业大学  
通讯作者：[李林](https://faculty.hzau.edu.cn/lilin12/zh_CN/index.htm)  
摘要：作者构建了包含基因组、转录组、翻译组、蛋白组的玉米整个发育时期的网络图谱。  
- [Abstract](#Abstract)
- [Introduction](#Introduction)
- [Result](#Result)
- [Method](#Method)
- [Fig](#Fig)
- 专业英语：
1. physical：adj.身体的，肉体的；物质的，有形的；外形的，外在的；物理的，物理学的；根据自然规律的，符合自然法则的；粗暴的，粗野的；性欲的，肉欲的
2. transitivity：n.传递性；动词的及物性；转移性
3. be preceded by: 在……之前发生
4. extent n.程度；范围，长度
5. axillary  腋生的；腋窝的；叶腋的
## 知识点：
### CRISPR-Cas9原理
- CRISPR-Cas9系统最初是在细菌中发现的一种免疫机制，用于抵御外来的病毒和质粒。CRISPR（Clustered Regularly Interspaced Short Palindromic Repeats）是一系列重复的DNA序列，它们之间由独特的间隔序列分隔。这些间隔序列是细菌在以前遭受病毒侵袭时获得的病毒DNA片段。Cas9（CRISPR-associated protein 9）是一种核酸酶，能够在特定的DNA序列上切割双链DNA。

- 在基因编辑中，研究人员设计了一个与目标DNA序列互补的单链RNA引导分子（sgRNA），将其与Cas9蛋白复合体一起导入目标细胞。sgRNA引导Cas9定位到目标DNA序列，并促使Cas9在该位置切割DNA双链。细胞修复切割产生的DNA断裂时，可以引入突变，从而实现基因的敲除、插入或替换。
### CRISPR-Cas9基因编辑方法
- CRISPR-Cas9基因编辑技术主要有两种应用方式：
1. 基因敲除（Knockout）：通过CRISPR-Cas9在目标基因上引入双链断裂（DSB），细胞通过非同源末端连接（NHEJ）修复机制修复断裂，过程中可能引入插入或缺失（indels）突变，导致目标基因功能丧失。
2. 基因插入或替换（Knockin）：利用同源重组（HDR）机制，在目标基因上引入特定的突变或外源序列。这需要提供一个含有所需改变的同源模板DNA，细胞利用该模板通过HDR修复DSB，实现精确的基因编辑
## Abstract {#Abstract}
>- Networks are powerful tools to uncover **functional roles of genes** in phenotypic variation at a system-wide scale.  
>- we constructed a maize network map that contains the **genomic, transcriptomic, translatomic and proteomic networks** across maize development.
>- We applied this map to identify **factors** regulating **flowering time** and identified **2,651 genes enriched in eight subnetworks**.
>- We validated the functions of 20 genes, including **18 with previously unknown connections** to flowering time in maize.
>- We uncovered a **flowering pathway involving histone modification**.
## Introduction {#Introduction}
>- Machine learning based on genome-wide epigenetic markers, informed by transcriptome and proteome data, was successfully used for the functional annotation of the maize (Zea mays L.) genome <sup>12</sup>
>> 12. Sartor, R. C., Noshay, J., Springer, N. M. & Briggs, S. P. Identification of the expressome by machine learning on omics data. Proc. Natl Acad. Sci. USA 116, 18119–18125 (2019).
## Result {#Result}
- [不同元件分子网络的特点](#1)
- [和网络分化相关复制的类型和年龄](#2)
- [玉米亚基因组可变网络的分化](#3)
- [重要基因的功能网络](#4)
- [预测开花时间新的功能基因](#5)

### <font color="#4169E1">**A landscape of molecular networks of different elements**</font> {#1}
>  为了组建大规模的玉米网络图谱，作者通过**双端标签测序结合染色体互作分析**整合了基因组网络、从31个不同组织和阶段的共表达分析鉴定到的所有元件网络、从21个组织和时期由翻译组学共表达分析鉴定的网络、贯穿玉米自交系B73的发育由酵母双杂RLL-Y2H方法鉴定的蛋白-蛋白外在互作、从而构建综合的基因组网络。
- To assemble a large-scale network map of maize, we integrated genomic networks identified by chromatin interaction analysis (ChIA) with paired-end-tag sequencing (PET)<sup>16</sup>;  
- networks for all elements detectable by coexpression analyses across 31 different tissues and stages;
- networks determined by translatomic coexpression analyses across 21 tissues and stages; 
- protein–protein physical interactions (Zea mays PPIs (ZmPPIs), interactome) identified by the yeast two-hybrid method RLL-Y2H17,18 across the development of the maize inbred line B73 (Supplementary Information, Supplementary Figs. 1–6, Extended Data Figs. 1–4 and Supplementary Tables 1–7). 
> 基于高质量数据，作者进行了一系列的多组学分析从而剖析网络中可检测到的功能元件和模块，其中有着潜在相似的功能的功能元件显著富集。多组学水平中，大约1.4k个模块中2.8百万的网络边缘由3w个基因组成。值得注意的是1.5w个蛋白编码基因形成了5.6w个高置信的玉米蛋白互作（相对精度78%；从36w个相对精度41%蛋白互作过滤而来）并形成1.1k个模块。尽管蛋白互作数据库离饱和还差很多，其仍然是植物中通过湿实验建立起来的大规模的蛋白互作数据集。
- Based on high-quality data, we conducted a series of multi-omic analyses to dissect networks of detectable functional elements and modules, in which functional elements with potentially similar roles were significantly enriched (Methods, Supplementary Information, Supplementary Figs. 1–6, Extended Data Figs. 1–4 and Supplementary Tables 1–7). 
- Approximately 30,000 genes formed over 2.8 million network edges in 1,412 modules at the multi-omics levels (Fig. 1 and Supplementary Table 6). 
- Notably, 15,476 protein-coding genes generated 56,243 high-confidence ZmPPIs (a relative precision of 78%) filtered from over 360,000 PPIs (a relative precision rate of 41%) and formed 1,147 modules. 
- Although the ZmPPI dataset is far from saturation (Supplementary Fig. 2), it is a large-scale PPI dataset obtained by wet lab experiments in plants.
> 多组学功能网络在拓扑结构展示出显著差异。基于不同的置信阈值，作者将网络分为低和高置信两种版本，这两种版本的网络有着相似的趋势并将高置信网络用于后续的分析。尽管来自不同网络的一系列中心节点存在重叠，三个维度的网络整体的传递系数是0.254，共表达网络是0.501,翻译网络是0.465，蛋白互作是0.05，突出了每个调控层不同的拓扑结构。
- The multi-omics functional networks exhibited markedly different topological hierarchies (Fig. 1a–d). 
- Based on different confidence cutoffs (Methods), we grouped the networks into low- and high-confidence versions, which showed similar trends (Supplementary Fig. 7). 
- We used the high-confidence networks for subsequent analyses. Although a subset of hub nodes (functional elements) from different omics data overlapped (Fig. 1e and Supplementary Fig. 7), the overall <font color="red">**transitivity coefficients（什么是传递系数？）**</font> were 0.254 in the three-dimensional networks, 0.501 in the coexpression network, 0.465 in the cotranslation network and 0.005 in the interactome (Supplementary Table 6), highlighting the distinct topological structure of each regulating layer (Fig. 1a–e).
> 为了缓解作者的单一网络非饱和的局限性并获得在时空分辨维度的所有可能的互作，我们整合了不同维度的网络。整体上，作者鉴定40w个功能元件并且建立起一个大于2.8百万边缘的综合的互作图谱。不同的调控原件（3.2w个mRNA、4k个长非编码RNA、152个小RNA、10个融合RNA和183个环状RNA）：例如作者检测到TGA1(玉米颖片结构)mRNA和一个长非编码RNA共表达相关性为0.851，一个环状RNA和一个小RNA相关性为0.846。作者并开发了一个用户友好的网站从而展示所有的表达和网络信息。总而言之综合的组学图谱为复杂植物网络提供了广泛的前景。
- To circumvent the limitations of nonsaturation in our individual networks and obtain all possible interactions in a temporally and spatially resolved manner, we integrated the networks from different layers. In all, we detected 407,721 functional elements and built an integrative interaction map with >2.8 million edges (Fig. 1f, Extended Data Fig. 1, Supplementary Fig. 8 and Supplementary Table 6). 
- Diverse regulating elements (32,758 mRNAs, 4,013 long noncoding RNAs (lncRNAs), 152 microRNAs (miRNAs), 10 fusion RNAs and 183 circular RNAs (circRNAs) were involved in this network): for example, we detected coexpression between teosinte glume architecture1 (TGA1, Zm00001d049822) mRNA<sup>19</sup> and one lncRNA (coexpression correlation coefficient R = 0.851), one circRNA (R = 0.848) and one miRNA (R = 0.846) (Fig. 1g). 
- We have developed a user-friendly website to host all the expression and network information (http://minteractome.ncpgr.cn/). 
- In summary, the integrative omics map provides a large-scale picture of a complex plant network.
### <font color="#4169E1">**Type and age of duplications related to network divergence**</font> {#2} 
> 基因功能的进化经常发生在全基因组加倍、串联复制、临近复制、或者由转座调节的复制和散在起源的复制。网络模块将基因参与相同过程的进行分组。<font color="red">**最短距离(SD)**(神马鬼，怎么计算的呀？)</font>是一种度量方式，代表网络中一对节点之间最短路径。当这一概念应用到复制基因中，较大的最短距离表明两个基因功能差异。所有的复制基因有着显著较高的属于相同模块的概率并且有着较低的SD值相较于非旁系同源基因。串联和临近重复基因最有可能富集在相同的模块，然而转座和散在分布的基因最不可能。源自全基因组加倍串联重复和临近重复的基因对有着最低的SD值，表明他们功能分化程度最低，然而源自转座和散在重复有着最高的SD值表明他们之间功能差异最大。
- Evolution of gene function is often preceded by whole-genome duplication (WGD), tandem duplication (tandem), proximal duplication (proximal) or duplications mediated by transpositions (transposed) and duplications of dispersed origin (dispersed) (Fig. 2a)<sup>20,21</sup>. Network modules group genes that potentially participate in the same process.
- The shortest distance (SD; Methods) is a metric that represents the shortest path between a pair of nodes within a network. When applied to duplicated genes, a large SD suggests that the two genes are functionally different. 
- All duplicated genes had a significantly higher probability of belonging to the same module (Fig. 2b) and had a lower SD (Fig. 2c) than non-paralogous genes. 
- Tandem and proximal genes were most likely to be enriched in the same network modules, whereas transposed and dispersed genes were least likely. 
- Gene pairs resulting from WGD, tandem and proximal duplications had the lowest SD values, indicating that they are the least functionally divergent, whereas those derived from transposed and dispersed duplications had the highest SD values, suggesting that they are the most functionally divergent (Fig. 2b,c).
> 为了在进化的角度，进一步探索复制基因对的网络分化，作者提出了两个指标：辛普森指数(SAB)和分化指数(DAB), 辛普森指数评估了两个节点间的相似性，然而DAB评估了两个基因的网络分化，并且这些指数共同作为功能分化很好的一个标志。基于SAB和DAB，作者将所有的复制基因分为5类分化模式，这五类在网络相似度有着明显的减少，两个基因间的差异显著增加从1类到5类。这五类模式潜在代表了不同类型的功能进化：1. 保守、2. 亚功能化、3. 新功能化和4. 非功能化或者特异化。值得注意的是整体的富集率从1到5逐渐减少<font color="red">**相较背景(背景是什么鬼？)**</font>，表明复制基因共有强烈的网络相似性而非偶然。不同类型的复制例如全基因组加倍和和串联重复有着相同的趋势。
- To further explore the network divergence of duplicated gene pairs from an evolutionary perspective, we introduced two indices, the Simpson index (SAB) and the divergence index (DAB). The SAB evaluates the similarity between two nodes, while the DAB measures the network divergence of two genes, and these indices are collectively a good indicator of functional divergence (Fig. 2d). 
- Based on SAB and DAB values, we classified all duplicated genes into five divergence patterns (Methods), with a clear gradual decrease seen in the extent of network similarity between the two genes as differentiation increased from type I to type V (Fig. 2d). These five patterns potentially represent various types of functional evolution: 
  1. conservation (type I)
  2. subfunctionalization (type II)
  3. neofunctionalization (types III and IV)
  4. nonfunctionalization/specialization (type V)<sup>22</sup>. 
- Notably, overall the enrichment ratios gradually decreased from type I to type V compared to background, indicating that duplicated genes share a stronger network similarity than would be expected by chance (Fig. 2e). Different types of duplications, such as WGD and tandem, showed the same trend (Fig. 2f and Extended Data Fig. 5). 
> 此外网络中不同类型的占比在遗传层面存在差异，有趣的是我们发现了基因复制后从共表达到共翻译再到互作水平的一个动态蜿蜒的网络。文章结果表明大于41%的复制在不同组学中是一致的，然而大约35%是和网络分化相关并且大约24%是和网络趋同相关。例如复制基因Zm00001d049815和Zm00001d031611从共表达到共翻译再到互作水平有着渐进的网络布线。较为重要的是，复制基因的比例起源于过去不同的时间，在1类到5类存在明显的差异，表明非功能化，新功能化和特异化很有可能和较古早的复制相关，然而功能保守很有可能和最近的复制相关。从网络分化到复制时间的这层关系变得越发的清晰，从共表达到共翻译再到互作组。总之，这些结果表明本文综合的网络图谱在进化的角度，包含潜在的和剖析基因功能有关的进化信息。
- Moreover, the proportions of different types of network divergence varied across genetic layers (Fig. 2g and Extended Data Fig. 5). Interestingly, we observed a dynamic rewiring of networks after gene duplication events from the coexpression to the cotranslation to the interactome level. 
- Our results indicate that approximately 41% of the duplicates were consistent across different omics layer, while approximately 35% were associated with network divergence and approximately 24% were associated with network convergence (Supplementary Fig. 9). 
- For example, the duplicated genes Zm00001d049815 and Zm00001d031611 displayed a progressive network rewiring from the coexpression to the cotranslation to the interactome level (Fig. 2h).
- Importantly, the proportions of duplicated genes originating at different times in the past varied dramatically from type I to V, suggesting that nonfunctionalization, neofunctionalization and specialization are more likely to be associated with older duplications, whereas functional conservation is more likely for the most recent duplications (Fig. 2i).
- This relationship of network divergence to duplication age became clearer from the coexpression to the cotranslation to the interactome. Together, these results indicate that our integrative network map encompasses potentially evolutionary information relevant to the dissection of gene function in an evolutionary perspective.
### <font color="#4169E1">**Variable network divergence of maize subgenomes**</font> {#3}
> 玉米是一个异源四倍体物种有着不同的古亚基因组，这里就定名为maize1和2，在基因含量和表达水平有着不对等的差异。尽管显性亚基因组玉米一相较玉米二有着更多的同源和高度表达基因，过去研究并没发现调控的偏好性可能是由于只在单一水平上评估了这一问题。因而，我们更详尽地剖析了两个亚基因组的调控偏好性，通过将所有亚基因组的基因分为两类：1. 玉米1基因中仍保留了玉米2的同源和玉米2基因有着玉米1基因的同源。作者随后比较了在不同组的基因的程度在不同组学水平上。鉴于此，作者计算了两个组基因程度的变异(方差)，并发现两个亚基因组在共表达水平没有显著差异，在共翻译水平差异增加了但是仍不够显著。然而，通过<font color="red">**程度值（又是什么鬼）**</font>在互作组水平有着显著差异，表明基因和节点的功能重要性。另外作者通过100个基因滑窗在源自高粱的古玉米亚基因组上定量了基因程度的整体差异并发现了玉米1中更多的显性区域从共表达到翻译组再到互作组水平，表明亚基因组网络连接性的差异在三个网络水平存在不同。
- Maize is an allotetraploid species with distinct ancient subgenomes, designated here as maize1 and maize2, which exhibit asymmetric divergence in both gene content and expression levels<sup>23</sup>. 
- Although maize1, the dominant subgenome, contains more orthologs and highly expressed genes than maize2, past studies did not find regulating bias, probably because they only assessed this question at a single level<sup>23,24</sup>.
- Therefore, we dissected the regulating bias of the two subgenomes in more detail by classifying all subgenome genes into two groups: maize1 genes with retained maize2 orthologs and maize2 genes with retained maize1 orthologs. 
- We then compared the degree (a network attribute; the number of edges of a node) of genes from each group at different omics levels. 
- To this end, we calculated the variation in the degree of genes from the two different groups. We detected no significant differences between the subgenomes at the coexpression level. The difference increased but was still not significant at the cotranslation level. 
- However, we observed a significant difference at the interactome level between the two subgenomes, as reflected by the degree values (Fig. 3a), which suggests the functional importance of genes and nodes<sup>25</sup>. 
- Additionally, we quantified the overall difference of the degree of genes in a sliding window (100 genes) in the ancient maize subgenomes derived from the Sorghum genome as previously published<sup>23</sup>.
- We detected more dominant bins for the maize1 subgenome from the coexpression to the cotranslation to the interactome level, suggesting that differences in network connectivity of the subgenomes vary across the three network levels (Fig. 3b and Supplementary Figs. 10 and 11).
### <font color="#4169E1">**Functional networks of important genes**</font>  {#4}
> 综合的网络图谱可以重构功能连接已知基因并发掘新的有着相似功能的调控基因。作者探索了涉及三个关键的分蘖基因涉及的亚网络：TB1，Gt1和Tru1。这三个基因的任何一个功能缺失都会导致古玉米的特点：更多分蘖的产生。这三个基因是带有ZmALOG1和ZmALOG2这两个基因的共表达网络的一部分，这俩个基因功能是未知的属于拟南芥LSH1和水稻G1转录因子家族的同源基因。值得注意的是这两个基因的通过Crispar介导的功能缺失也会导致更多的分蘖和那三个基因的突变功能类似。另外酵母双杂表明这两个基因可以和TB1互作。这些结果阐明在亚网络内的已知和未知基因网络边缘有着潜在的生物学意义。因而该文的玉米综合图谱可以用来产出对感兴趣基因功能的假设以及推断的网络通路。
- The integrative network map can reconstruct the functional connections of well-known genes and uncover new regulating genes with similar functions (Fig. 4).
- We explored a subnetwork involving three key tillering genes, teosinte branched 1 (Tb1, Zm00001d033673), grassy tillers 1 (Gt1, Zm00001d028129)<sup>27</sup> and tassels replace upper ears 1(Tru1, Zm00001d042111).
- Loss-of-function mutations in any of these genes lead to the formation of more tillers, a characteristic of the maize ancestor teosinte.
- Tb1, Gt1 and Tru1 were all part of a coexpression network with ZmALOG1 (Zm00001d003057) and ZmALOG2 (Zm00001d032696), two genes of unknown function belonging to the Arabidopsis LSH1 and Oryza G1 (ALOG) transcription factor family (Fig.4a). Notably, the loss-of-function mutations in ZmALOG1 and ZmALOG2 obtained by CRISPR- or CRISPR–Cas9-mediated gene editing led to higher tillering, similar to the tb1, gt1 and tru1 mutants (Fig. 4b and Extended Data Fig. 6a). Moreover, yeast two-hybrid assays indicated that ZmALOG1 and ZmALOG2 can interact with TB1 (Fig. 4c). These results demonstrate that the network edges between known and unknown genes in subnetworks have potential biological meaning.
- Therefore, our maize network map can be used to generate hypotheses about the functions of genes of interest, as well as their putative network pathways.
> 综合的网络图谱包含具有潜在相似遗传功能的功能亚网络。杯装子叶基因建立起很多植物物种器官间的边界。然而玉米中的CUC基因却没有很好的验证。为了测试玉米网络图谱用于挖掘感兴趣基因功能的适应性，我们从亚基因组中查找到CUC基因（ZmCuC，ZmNAM1，ZmNAM2），并发现这三个基因和Tsh1、Ba1、PIN1互作。在玉米和其他物种中呢，这些基因编码的蛋白影响侧枝器官的形成和发育。因而，作者探寻了这些CUC模式的基因在侧枝器官发育的可能功能。实际上，Crispar介导的Nam1和Nam2的单突或者双突功能缺失会导致腋芽形成的缺陷,从而确定这些基因的产物通过建立边界在侧器官的形成有着保守的功能，并为侧芽分生组织的形成所需。CUC基因和其他基因在亚网络中的互作，包括Pin1、Tsh1和未知基因bHLH168和MADs36是未来研究的候选。
- The integrative map contains functional subnetworks with molecular relationships for genes of potentially similar genetic functions (Fig. 4d). 
- CUP-SHAPED COTYLEDON(杯状子叶) (CUC) genes establish boundaries between organs in many plant species<sup>35</sup>. However, the roles of CUC genes in maize have not been well characterized. To test the suitability of the maize network map for discovering the functions of genes of interest, we searched the subnetworks for the maize CUC genes (ZmCUC3, Zm00001d031463; ZmNAM1, Zm00001d035266; ZmNAM2, Zm00001d041791); these three genes interacted with tassel sheath1 (Tsh1, Zm00001d039113), barren stalk1 (Ba1, Zm00001d042989) and pin-formed1 (ZmPIN1, Zm00001d044812) (Fig. 4d). The proteins encoded by these genes (or their homologs) function in lateral organ formation and development in maize or other species<sup>28–34</sup>. 
- Thus, we reasoned that these maize CUC-like genes might have roles in lateral organ development. Indeed, loss-of-function zmnam1 and zmnam2 single and double mutants generated by CRISPR–Cas9-mediated gene editing were defective in lateral bud formation (Fig. 4e and Extended Data Fig. 6b), confirming that the products of these genes have conserved roles in lateral organ formation by establishing boundaries, which are required for proper lateral meristem initiation. 
- The interactions between the ZmCUC genes and other genes in the subnetwork, including Pin1, Tsh1 and the unknown genes basic helix-loop-helix 168 (bHLH168, Zm00001d003806) and MADS36 (Zm00001d043589), are candidates for further research.
> 综合的网络图谱也可以阐明网络基因的分子调控关系，并分别克隆用于复杂性状。玉米粒尺寸和重量是玉米产量重要的组成并广泛研究，总共63个已知玉米粒突变基因已经被功能克隆，我们鉴定了综合网络图谱中63个中的62个基因并随机分为两组。62个基因中，我们将45个基因用于玉米粒亚网络的提取，此外作者保留了剩余的17个基因作为测试集从而评估它们是否显著富集在提取的玉米粒相关的亚网络而且验证了玉米粒相关的调控网络的鲁棒性。携有45个随机选择的玉米粒基因的亚网络(1k次模拟)成功的预测了平均52%的验证了的玉米粒基因(最高到82%)，这个值显著高于由随机选择和功能无关的基因所构建的控制子网络。值得注意的是，所有的62个玉米粒基因显著装配在一个联系子网中，并分为6个子模块。每个模块内的基因显著富集在胚发育、淀粉代谢、糖调控的信号通路和代谢过程。这些基因中，一个编码三角状五肽重复含量蛋白并先前未报道的基因通过Crispr编辑的突变会导致玉米粒的皱缩。通过玉米网络图谱，作者也成功的重构了调控网络涉及淀粉合成通路和其他建立的通路，因而这个玉米网络图谱是一个强有力的工具用于探索和组装在复杂性状下的基因调控网络。
- The integrative network map can also illustrate the molecular regulating relationships of genes that were cloned separately for complex traits. 
- Maize kernel size and weight are important components of maize yield and have been investigated intensively. A total of 63 well-known kernel mutant genes have been functionally cloned (Supplementary Table 8). We identified 62 of these 63 genes in the integrative network map and randomly divided them into two groups. 
- Of the 62 genes, we used 45 genes for the extraction of kernel-related subnetworks, while we retained the remaining 17 genes as a test set by assessing whether they were significantly enriched in the extracted kernel-related subnetworks and thus validate the robustness of the kernel-related regulating networks extracted from our integrative network map. 
- The subnetworks obtained with the 45 randomly selected kernel genes (1,000 simulations) successfully predicted an average of 52% of the validated kernel genes (up to 82%), a value significantly higher (P < 1.0 × 10−16) than that for control subnetworks constructed with randomly selected and functionally unrelated genes (Supplementary Fig. 12a). 
- Notably, all 62 kernel genes were significantly (P < 1.0 × 10−16) assembled into one linked subnetwork (more than random) (Fig. 4f), which could be divided into six submodules. The genes in each submodule were significantly enriched in the functional groups of embryo development, starch metabolic process, sugar-mediated signaling pathway and metabolic process. 
- Among these genes, the mutation via CRISPR–Cas9 gene editing of a previously unreported gene, Zm00001d003047, encoding a pentatricopeptide(三角状五肽) repeat-containing (PPR) protein, led to shrunken kernels (Fig. 4f–h and Supplementary Fig. 12b). 
- Using the maize network map, we also successfully reconstructed regulating networks involved in the starch synthesis pathway36–39 and other established pathways (Extended Data Fig. 7 and Supplementary Tables 9 and 10). 
- Thus, the maize network map is a powerful tool for exploring and assembling gene regulating networks underlying complex traits.
### <font color="#4169E1">**Predicting novel functional genes for FT**</font> {#5}
> 开花时间是重要的农艺性状，涉及玉米在全球广泛气候下适应性。尽管玉米中有很多开花时间相关的基因已经被克隆了，但很多仍然需要了解其中的遗传调控和分子机理。为了探索潜在的调控网络，作者基于综合的网络图谱和机器学习(用已知的FT基因：玉米中或者FT同源基因在其他物种中功能克隆作为训练集)构建了预测模型。这个模型展现出高度的预测精度并且不同互作图谱层有着不同的预测能力，其AUC值从0.67~0.93分布。作者预测了2.6k个与开花时间相关的基因，反映出玉米开花基因中复杂的分子机理。作者评估了已经验证的和预测的基因间的SD值。预测的FT基因和通过背景基因获得的基因有着显著较低的SD值
- FT is an important agronomic trait involved in the adaptation of maize to a wide range of climates worldwide40–42.
- Although many key FT genes in maize have been cloned (Supplementary Table 11), much remains to be learned about the genetic control and molecular mechanisms underlying this trait.
- To explore the potential regulating networks,we constructed a prediction model based on the integrative map and a machine learning method using known FT genes (functionally cloned FT genes in maize and maize homologs of FT genes cloned in other species) as the training dataset (Methods, Extended Data Fig.8 and Supplementary Figs.13 and 14).
- The model showed high prediction accuracy and different layers of the interaction map had variable prediction power, with area under the curve (AUC) values ranging from 0.67 to 0.93 (Fig.5a and Supplementary Fig.14).
- We predicted that 2,651 genes are associated with FT (Supplementary Table 12), reflecting the highly complex molecular mechanism underling FT in maize.
- We assessed the SD between each known validated FT gene and all predicted FT genes.
- The predicted FT genes had significantly lower SD values (to well-known FT genes) compared to those obtained against background genes (Fig.5b).
> 相较于背景基因，预测的FT基因显著富集在拟南芥同源数据集和FT的全基因组关联分析研究。此外，基于之前的基因调控模型和我们的综合网络，作者鉴定出8个开花时间相关的调控模块。
- Compared to background genes, the predicted FT genes were significantly enriched both in the Arabidopsis homology dataset and in a genome-wide association study of single-nucleotide polymorphisms43 associated with FT(Extended Data Fig.9a).
- Furthermore, based on a previous gene regulatory model<sup>44</sup> and our integrative network, we identified **eight molecular subnetworks associated with FT** in maize: 
  1. the light transduction pathway (光转导途径)
  2. circadian clock (昼夜节律)
  3. the photoperiod pathway
  4. the autonomous pathway (自主通路)
  5. the gibberellin (GA) pathway
  6. a pathway for the maintenance of inflorescence meristem identity (integrator)
  7. the floral transition pathway (花的转化通路)
  8. an unknown pathway including genes with different predicted functions (Fig.5c, Extended Data Fig.9b and Supplementary Table 13).
> 作为测试，作者敲除了ZmPRR95a基因(拟南芥昼夜戒律组成假响应调控9).作者评估了三个FT性状突变（DTA：到开花时间；DTT：抽穗时间；DTS：抽丝时间）。突变系携有两个编辑等位基因任何其中一个的表现出开花时间的显著延后（云南2018年夏天和吉林2018年的夏天）相较于野生型。作者也从一个未知开花时间通路中敲除了编码一个MYST组蛋白乙酰转移酶的Zm00001d006918基因，发现两个独立的敲除系相较野生姐妹系开花较晚。总共，作者确定了20个基因（其中7个预测的基因在仍和植物中都没有功能的证据，是完全的新基因，11个是没在玉米中验证但在其他物种中的同源是影响开花时间的，两个已知玉米开花时间遗传基因，这些都在两个及其以上的大田环境或者极度差异的环境中对野性型和突变型做了对照实验）。
- As a test, we knocked out one gene, Zm00001d021291 (ZmPRR95a, a homolog of the Arabidopsis circadian clock component PSEUDO-RESPONSE REGULATOR 9 (PRR9, AT2G46790))45, via CRISPR–Cas9-mediated gene editing (Fig.5d, Extended Data Fig.9 and Supplementary Fig.15).
- We evaluated the mutants in the field for three FT-related traits (DTA, days to anthesis; DTT, days to tasseling; DTS, days to silking).
- Mutant lines carrying either of two edited alleles displayed a significant delay in their FT in Yunnan in the summer of 2018 and in Jilin in the summer of 2019 compared to their wild-type (WT) siblings (Fig.5e and Supplementary Table 13).
- We also generated a knockout for Zm00001d006918, encoding a MYST histone acetyl transferase46, from an unknown FT pathway, via CRISPR–Cas9-mediated gene editing.
- We determined that two independent knockout lines flowered later than their WT siblings (Fig.5f–h).
- In total, we confirmed 20 genes (of which 7 predicted FT genes were completely new genes without evidence for a function in FT in any plant species, 11 predicted FT genes were not validated genes in maize but had homologs in other species affecting FT and 2 were well-known FT maize genes associated with FT when comparing the FT of their respective mutant and WT lines either in two or more field environments or with an extremely significant difference in one environment) (Fig.5d–h, Supplementary Figs.15–33 and Supplementary Tables 13 and 14).
>综合来说，所有的这些突变体实验的敲除表明预测开花时间高度的鲁棒性通过使用综合的网络图谱进行机器学习。20个验证的基因提供了理想的基因资源来提升对于玉米开花时间调控网络的理解。例如5个基因是和昼夜节律相关，包括一个从未被鉴定过的基因其，编码B-box锌指蛋白的Zm00001d006198。有趣的是，作者也挖掘了一个未知的开花时间通路，包含8个基因，其中的Zm00001d006918编码组蛋白乙酰转运酶MYST家族101成员。这些基因在拟南芥中的同源调控春化响应基因FLC通过表观遗传修饰（H4K5ace，反义RNA合成）和核复合物的形成。尽管春化在玉米中是不存在的，玉米基因中在拟南芥中的同源其在春化的功能提供了玉米开花通路的证据。始料不及的挖掘未知复杂的玉米开花时间通路在玉米未来驯化中为研究开花时间和春化提供了基础。鉴于此，作者系统挖掘并验证了一些列的开花时间分子通路，作者探究来自于这些通路的基因是否可能对开花时间的贡献具有一致的形式。生长在相同环境的开花时间功能缺失突变，展现出对开花时间广泛并显著的影响。
有趣的是，来自相同通路的这些功能缺失突变基因对于开花时间有着一致的遗传效应，表明复杂的调控网络机理影响着玉米开花时间。
- Together, the results of all these knockout mutagenesis experiments suggest a high robustness for the prediction of FT genes by machine learning using our integrative network map.
- The 20 validated genes provided an ideal gene resource to advance our understanding of the regulating network underlying maize FT by exploiting our integrative network map (Fig.5d–h, Extended Data Fig.9 and Supplementary Figs.15–33).
- For example, five genes were associated with the circadian clock, including Zm00001d006198 (encoding a B-box zinc finger protein; Supplementary Fig.16), a gene that had not been identified in previous studies.
- Interestingly, we also uncovered an unknown FT pathway consisting of eight genes, including Zm00001d006918 (encoding a histone acetyl transferase MYST family 101 member; Fig.5f–h, Extended Data Fig.9 and Supplementary Fig.15b), Zm00001d028173 (FRIGIDA-like protein 1) and others (Extended Data Fig.9 and Supplementary Figs.17–21).
- Homologs of these genes in Arabidopsis regulate the vernalization response gene FLOWERING LOCUS C (FLC, AT5G10140) via epigenetic modification (H4K5ace<sup>47</sup>; antisense RNA synthesis<sup>48</sup>) and nuclear complex formation49.
- Although vernalization does not occur in maize, maize genes whose homologs in Arabidopsis function in vernalization showed evidence of function in maize FT pathways.
- The unexpected discovery of this unknown complex FT pathway in maize lays the foundation for studying the relationship between FT and vernalization during maize domestication in the future24,40.
- Given that we systematically uncovered and validated a series of molecular pathways underlying FT in maize, we asked whether genes from these pathways might contribute to FT in a consistent fashion.
- Loss-of-function mutants of FT genes grown in the same environment (Shandong, China in the summer of 2020) exhibited a wide range of significant effects on the timing of flowering (Supplementary Fig.34 and Supplementary Table 13).
- Interestingly, loss-of-function mutations of genes from the same pathway had inconsistent genetic effects on FT (Supplementary Fig.34), suggesting a complex regulatory mechanism influencing FT in maize.

## Method {#Method}
### Plant materials 
Thirty-one tissues or stages (Supplementary Table 1) were sampled from B73 maize plants grown in a greenhouse.
- The plants were grown under a 12-h light–12-h dark photoperiod at 25 °C.
- Two biological replicates were collected per tissue.
- Total RNA was extracted from 5–10 individual plants per replicate.
- Two biological replicates were separately used for sequencing.
- Notably, the transcriptomic and translatomic datasets of five tissues (Root-14DAS, Seedling-14DAS, Stem-14DAS, Leaf-14DAS, Tassel-V12) were generated in our previous study54 (see ‘Data availability’).
- Detailed bioinformatics scripts and the resulting data have been deposited in GitHub (https://github.com/hanlinqian/IntegrativeNetworkMap) and a bioinformatics analysis flowchart can be found in Extended Data Fig.10.
### RNA-sequencing analysis 
#### RNA isolation, library construction and sequencing 
- The processes of RNA isolation, library construction and sequencing are described in the Supplementary Information.
#### Quality control of raw data 
- BGI used SOAPnuke2.1.6 (ref.55) to process the raw RNA-sequencing (RNA-seq) and circRNA-seq data with the parameters -n 0.1 -q 0.5 -l 12.
- Low-quality raw sRNA-seq reads were filtered and adaptors were removed using cutadapt v.1.9.1 (https://cutadapt.readthedocs.io/en/stable/) with the parameters -q 20,20 and -a AGTCGGAGGCCAAGCGGTCTTAGGAA.
#### Identification and quantification of total RNA 
- Clean RNA-seq reads from the 31 tissues or stages were mapped to unique locations in the B73 AGPv4 reference genome using the default parameters of HISAT2 (2.1.0)56.
- The mapping and quantification of the transcriptome were conducted using StringTie (v.1.3.0-foss-2016b)57 with the parameter -G -e -A.
#### Identification and quantification of lncRNAs 
- The improved version of LncRNA_Finder<sup>58</sup> was used to identify lncRNAs from total RNA.
- Transcripts longer than >300 nucleotides were removed because >95% of the protein-coding genes encode proteins with >100 amino acids<sup>59</sup>.
- The remaining transcripts were used as an input for CPC2 (ref.60), which rapidly and accurately assesses the coding ability of RNAs to discriminate and remove potential protein-coding transcripts.
- To remove housekeeping RNAs, including [tRNAs](http://gtrnadb.ucsc.edu), [rRNAs](http://maize.jcvi.org/), [snRNAs and snoRNAs](http://noncode.org/); the remaining transcripts were compared to the listed housekeeping RNA databases.
- The remaining transcripts were considered to be lncRNAs.
- StringTie<sup>57</sup> was used to quantify lncRNA expression levels with the parameters -G -e --A.
#### Identification and quantification of fusion RNAs 
- Mapsplice (v.2.1.8)61 was used to identify fusion RNAs from total RNA-seq data from each tissue.
- The following filtering steps were performed to obtain high-confidence fusion RNAs.
- First, poly(A) sequences were removed using the following criteria: the proportion of As and Ts was taken as 40% and the proportion of As and Ts was taken as >80%.
- Second, to remove fusion RNAs formed by homologous genes, the two parental genes (overlapping sequences of >50 bp) that formed the fusion RNA were identified, and the similarity between the parental genes was monitored.
- If no corresponding parental genes were found (overlapping sequences of <50 bp), the sequences were extended 500 bp upstream and downstream of the splicing site, and the extended sequences were aligned to the maize AGPv4 reference genome to identify the parental genes.
- The remaining fusion RNAs were compared to the genome with BLAST (v.2.2.18)62, and the number of bases overlapping the splicing site was manually checked.
- Only fusion RNAs with <7 overlapping bases were retained.
- Third, because some fusion RNAs with high sequence identity might be the same fusion RNAs, BLAST62 analysis was performed to divide the highly similar fusion RNAs into groups and the fusion RNAs with the highest number of reads in each group were retained.
- Fourth, to validate the identified fusion RNAs, their sequences were compared to their original sequences using TopHat (v.2.1.1)63, and RNAs that matched at least six bases and had a read coverage of at least three at the junction site were retained.
- The normalized expression levels of fusion RNAs were calculated by the number of reads mapped to a fusion junction locus from a given tissue divided by all the reads mapped.
#### circRNA identification and quantification 
- To obtain clean reads from circRNA-seq, FastUniq (v.1.0)64 was used to remove duplicates introduced by PCR amplification during high-throughput sequencing.
- CIRCexplorer2 (v.2.0.1)65 was used to align and quantify the expression levels of the circRNAs.
- The mean counts of junction reads per mapped billion reads (CPM) values were calculated as the normalized expression level of each transcript.
- The CPM formula for circRNAs is as follows:  where Readsj is the number of junction reads and Readsm is the number of mapped reads.
#### sRNA identification and quantification 
- sRNAs were identified and quantified as described by Wang et al.66 with slight modifications.
- After removing low-quality reads and eliminating those matching tRNAs, rRNAs, snRNAs and snoRNAs, the retained clean reads from 21 tissues or stages that were 18–26 nucleotides in size were mapped to the maize AGPv4 reference genome using Bowtie (v.1.1.2) with parameters of unique location and no mismatch67,68,69,70.
- All samples were merged and the read number was calculated using SAMtools (v.1.5)71; only sRNAs with more than 6× coverage were retained.
- Given that sRNAs are distributed in clusters throughout the maize genome and that gap distance is defined as the distance between two neighboring clusters, we compared the coverage of chromosomes between two neighboring sRNA clusters using different gap distances and selected a coverage of 6× and a 300-bp gap distance as the thresholds for identifying sRNAs.
- For each sample, the number of sRNA reads in each sRNA cluster was calculated and reconstructed as a matrix, with each row representing an sRNA cluster and each column representing a sample.
- Read abundances were normalized based on library size by scaling to transcript per million (TPM) to allow direct comparisons across libraries.
### Ribo-sequencing: preparation, sequencing and analysis 
- The methods for ribosome profiling and sequencing are described in the Supplementary Information.
- Two high-quality biological replicates of 21 tissues or stages were subjected to Ribo-sequencing (Ribo-seq) in our study.
- After removal of adaptors and quality control by SOAPnuke (v.2.1.6)55, the Ribo-seq reads were aligned to the rRNA reference sequence from the NCBI using Bowtie (1.1.2)72 to remove reads derived from rRNAs.
- Then, cleaned Ribo-seq reads were mapped to the maize AGPv4 reference genome68.
- Expression levels were measured with HISAT and StringTie56,57,67.
- Only reads that mapped to unique coding sequences were used to calculate translational abundance.
### PPIs 
#### PCR amplification 
- Total RNA was extracted from 20 tissues separately using a Direct-zol RNA Miniprep kit (cat.no.R2061, Zymo Research).
- Total RNA was divided into eight groups (Supplementary Table 2) based on tissue type.
- Equal amounts of RNA from each group were pooled.
- Approximately 2 μg pooled RNA was used to generate first-strand complementary DNA (cDNA) using Clontech’s SMART technology73 according to Clontech’s Make Your Own Mate & Plate Library System User Manual (protocol no.PT4085-1) with modified primers.
- These modified primers were applied to modified pGADT7 (mAD) or modified pGBKT7 (mBD) vectors from a previously reported recombination-based library versus a library high-throughput yeast two-hybrid (RLL-Y2H) screening system17 and included modified SMART III oligonucleotide primers (Supplementary Table 15) and modified oligo(dT) primers (Supplementary Table 15).
- The first-strand cDNA was amplified to obtain full-length double-stranded cDNA using a KAPA HiFi HotStart ReadyMix PCR kit (cat.
- no.
- KK2631, KAPA) with specific primers (for mAD: mAD-F and mAD-R; for mBD: mBD-F and mBD-R; Supplementary Table 15).
- Next, the abundance of full-length double-stranded cDNA was normalized as described previously74 and 10 μg of normalized cDNA was used for each yeast transformation.
#### Yeast library construction 
- Modified mAD and mBD vectors were used to construct a maize cDNA yeast library according to the Clontech Y2H transformation method.
- The mAD and mBD vectors were digested with the restriction enzymes BamHI and NdeI.
- Each 10 µg of PCR-amplified cDNA fragment was mixed with 5 µg of linearized mAD or mBD vector and used to cotransform 6 ml of Y187 or Y2H competent yeast cells, respectively, with the Yeastmaker Yeast Transformation System (cat.
- nos.
- 630439, 630489 and 630490, Clontech) according to the manufacturer’s protocol.
- The cells were plated onto synthetic defined medium without leucine (mAD clones) or synthetic defined medium without tryptophan + X-α-galactose (mBD clones) and grown for 3–4 d at 30 °C.
- All blue clones (generated by self-activation) in the BD library were removed.
#### Yeast mating, plasmid isolation and PPI sequencing 
- The processes of yeast mating, plasmid isolation and PPI sequencing are described in the Supplementary Information.
#### Removal of auto-activators in the Y2H screens 
- We also performed yeast mating between a yeast strain harboring the empty AD vector and yeast strains containing cDNA libraries (with each clone fused to the GAL4 DNA-binding domain).
- A few surviving clones on synthetic defined medium without tryptophan, leucine, histidine and adenine were obtained and collected for PacBio sequencing.
- The genes corresponding to these clones were considered as auto-activators.
- Any PPIs involving these auto-activators were removed from our high-confidence ZmPPI dataset.
#### Bimolecular fluorescence complementation assays 
- Details of these assays are described in the Supplementary Information.
#### PPI analysis 
- The libraries were sequenced on the PacBio platform, and circular consensus sequencing (CCS) reads were called using SMRTLink (v.5.1.0.26412) (https://www.pacb.com/support/software-downloads/).
- The following analysis was performed to obtain PPIs.
- First, reads in FASTA format were extracted from BAM files using SAMtools (v.1.9)71.
- Next, because ten barcoded samples were combined to generate each sequencing library, reads were demultiplexed into individual libraries based on their barcode sequences using cutadapt (v.1.9.1)75.
- The parameters -e and -m for running read separation were 0.3 and 200, respectively.
- The reads were aligned to the coding sequence data from the maize AGPv4 reference genome to identify candidate PPI genes using BLAST+ (v.2.7.1)62.
- Only PPI genes with a BLAST e-value <1.0 × 10−4 that matched two genes were retained.
- Lastly, the mAD-primer, mBD-primer and recombination sequence (Supplementary Table 15) were aligned to the retained CCS reads using needle (http://biopython.org/DIST/docs/tutorial/Tutorial.html#htoc86).
- If a CCS contained sequences from two different genes, and the primer and recombination sequences were aligned consistently, the proteins encoded by the two genes were considered to be interacting proteins.
-  The PPIs were divided into three confidence groups based on the probability of self-activation.
- Low-confidence PPIs indicate interaction edges detected using offline data.
- Because the mating of yeast prey libraries with empty bait (BD) vector generated interaction signals (self-activation), to remove the edges caused by self-activation, all PPIs involving eight empty BD libraries were removed to obtain medium-confidence PPIs.
- To further remove false PPIs introduced by self-activation, PPIs identified using empty vectors were used as the positive training data, true PPIs confirmed by bimolecular fluorescence complementation were used as the negative training data and the frequencies of each gene in the AD and BD libraries were used as training features to estimate the self-activating genes using the M5P machine learning classification algorithm (https://cran.r-project.org/web/packages/RWeka/index.html).
- Edges for genes producing self-activation in the BD library were removed, and the remaining PPIs were considered to be high-confidence PPIs.
- The saturation detection in each library used the nls function from the R package, with the parameters Vm = 30,000, K = 30′ (Supplementary Fig.2).
### Construction of coexpression networks 
- An expression matrix of 37,180 elements (31,294 genes, 5,427 lncRNAs, 12 fusion RNAs, 289 circRNAs and 158 miRNAs) detected in more than 10 distinct tissues or stages was used to construct a transcriptome coexpression network.
- WGCNA (v.1.70-3)76 was used to construct the coexpression networks with default parameters.
- Slim coexpression networks were constructed using 30,963 annotated protein-coding genes, and a slim coexpression network was used in all functional analyses.
### Construction of an integrative omics database 
- Four omic datasets were used to construct an integrative multi-omics database including all possible interactions: 
  1. ChIA–PET data from Peng et al.14
  2. a coexpression network constructed using RNA-seq data from 31 tissues or stages
  3. a cotranslation network constructed using Ribo-seq data from 21 tissues or stages
  4. PPIs from yeast two-hybrid screening.
- The weight of the integrative omics map was equal to the sum of the weights of different levels.

  

## Fig {#Fig} 
> [Fig1](https://www.nature.com/articles/s41588-022-01262-1/figures/1)  
> [Fig2](https://www.nature.com/articles/s41588-022-01262-1/figures/2)  
> [Fig3](https://www.nature.com/articles/s41588-022-01262-1/figures/3)   
> [Fig4](https://www.nature.com/articles/s41588-022-01262-1/figures/4)   
> [Fig5](https://www.nature.com/articles/s41588-022-01262-1/figures/5)   


## Pipeline

### Section0
>这一部分是鉴定和定量mRNA、lncRNA、circRNAs和小RNA的，通过Ribo-seq计算蛋白编码RNA表达量并获得蛋白-蛋白互作
>>This section is about identifying and quantifing mRNAs, lncRNAs, circRNAs and small RNAs; calculating protein-coding RNA expression by Ribo-seq; and obtaining protein-protein interactions.
#### Pipeline-circrna: Pipeline of identifying and quantifing circRNAs by circRNA-seq
> 通过SOAPnuke和FastUniq预先处理原始数据获得的清洗数据被提交到了GSA网站 (https://ngdc.cncb.ac.cn/gsa/, accession number: CRA006221)上了，最后的输出表达量文件和注释文件在section0/data路径下。
>> The clean data obtained after preliminary processing of the original data by SOAPnuke and FastUniq software was submitted to GSA(https://ngdc.cncb.ac.cn/gsa/, accession number: CRA006221). Final output expression file (21tis-circ.tpm) and annotation file (circ.gff3) were uploaded into section0/data.
- step1: build genome index  建立基因组索引
```sh
module load Bowtie/1.2.2; bowtie-build Zea_mays.AGPv4.dna.toplevel.fa bowtie-1.2.2-index/Zea_mays.AGPv4.dna.toplevel
```
- step2: map to genome by TopHat 通过TopHat比对到基因组
```sh
module load Tophat/2.1.1; module load Python/2.7.15; module load Bowtie/1.1.2-foss-2016b;  
mkdir 01CIRCexplorer2/$i;cd 01CIRCexplorer2/$i;
tophat2 -o fusion-$i -p 10 --fusion-search --keep-fasta-order --bowtie1 --no-coverage-search Zea_mays.AGPv4.dna.toplevel uniq-$i-1.fq.gz uniq-$i-2.fq.gz;
```
- step3: identify circRNA and calculate read count by CIRCexplorer2 通过CIRCexplorer2鉴定circRNA并计算read数
```sh
module load SAMtools/1.9; samtools index -b tophat-fusion/fusion-$i/accepted_hits.bam ##build bam index
CIRCexplorer2 parse --pe -t TopHat-Fusion fusion-$i/accepted_hits.bam > CIRCexplorer2_parse-$i.log;
CIRCexplorer2 annotate -r b734.40/4.40.gpd -g Zea_mays.AGPv4.dna.toplevel.fa -b back_spliced_junction.bed -o circularRNA_known-$i.txt; ##start position of bed file is 0 while gff3 file is 1
```
- step4: merge all the tissues' read count to one file  合并所有的组织read数到一个文件里
```sh
perl mergetpm.pl > 21+21circrna.tpm ##mergetpm.pl refers to Section0/code/mergetpm.pl with slight modifications
```  
- step5: transfer read count to TPM(CPM)  
  > 根据CPM的计算公式，CPM=A/mapped reads*1000000 ,A指的是比对到基因的read数
  >>According to the formula of CPM, CPM= A/mapped reads*1000000 (A is the read count mapped to a gene).

#### Pipeline-lncrna: Pipeline of identifying and quantifing lncRNAs
> 步骤一中的bam文件在totalRNA.md中.最后的表达文件和注释文件在section0/data
>> The input bam files in step1 of this pipeline are from step2 of pipeline-totalrna.md in section0. Final expression file (21tis-lnc.tpm) and annotation file (lncrna.gff3) were uploaded into section0/data.
- step1: obtain new annotation sites  
```sh
module load StringTie/1.3.0-foss-2016b; stringtie hisat+rf/samtools_$i\_sort_uq.bam -o GTF/$i\.gtf -p 2 -G Zea_mays.AGPv4.40.gff3 -l $i
```
- step2: merge all the gtf files by Cufflinks
```sh
cuffmerge -p 8 -g Zea_mays.AGPv4.40.gff3 list.txt -o merged-cuffmerge.gtf
stringtie --merge -G Zea_mays.AGPv4.40.gff3 -m 100 -c 0.1 -f 0.05 -o merged-stringtie.gtf list.txt
```  
- step3: screen new transcript loci with class code "u" and the number of exon exceeds 1  
- step4: transfer gtf file to fa file  
```sh
gffread -w merged-select.fa -g Zea_mays.AGPv4.dna.toplevel.fa merged-select.gtf
```  
- step5: identify lncRNAs  
```sh
module load Bowtie2/2.4.4; module load BLAST+/2.7.1; module unload Python/3.6.5; module load Python/2.7.15; module load CPC2/beta; perl LncRNA_Finder2.0.pl -i merged-select.fa -k zmnc-idsimple.fa -o newlnc
```  
- step6: calculate expression of lncRNAs  
```sh
module load StringTie/1.3.0-foss-2016b; stringtie hisat+rf/samtools_$i\_sort_uq.bam -p 3 -G alllincrna.gtf -e -A exp/$i_linc_exp.tab
``` 
- step7: merge all the tissues' expression to one file
```sh
perl mergetpm.pl > 21+21lncrna.tpm #The code refers to mergetpm.pl in the code folder, and slightly modified according to the array location.
```  

#### Pipeline-ppi: Pipeline of obtaining protein-protein interactions of maize after RLL-Y2H assays
> 原始数据在[GSA数据库](https://ngdc.cncb.ac.cn/gsa/),CRA006221。最后的输出互作文件edge-proteome-lowconf.txt在section0/data
>> Raw data was submitted to GSA database (https://ngdc.cncb.ac.cn/gsa/) with accession number CRA006221. Final output interaction file (edge-proteome-lowconf.txt) were uploaded into section0/data. 
- step1: bamtofq  
```sh
module load SMRTLink/9.0.0.92188; ccs test.subreads.bam test.fastq.gz ‐‐report‐file test_report.txt ‐‐min‐length 50 ‐‐max‐length 100000 ‐j 20
```
- step2: fqtofa  
```sh
seqkit fq2fa test.fastq.gz ‐w 0 >test.fa
```
- step3: Distinguish sequences from AD and BD vector according to ATTL sequence  
```sh
module load BLAST+/2.6.0;  blastn ‐db attl_blastindex/attl_db ‐query test.fa.fa ‐out testattlblast.txt ‐evalue 1.0e‐4 ‐outfmt 6;
perl attl.pl test‐attlblast.txt test.fa BD‐test.fa AD‐test.fa ##attl.pl was in Section0/code
```
- step4: Sequences were mapped to CDS region  
```sh 
module load BLAST+/2.6.0; makeblastdb ‐in Zea_mays.B73_RefGen_v4.cds.all.fa ‐dbtype nucl ‐parse_seqids ‐out blast‐v4.cds.all.fa;
blastn ‐db blast‐v4.cds.all.fa ‐query BD‐test.fa ‐out BD‐test2cdsblast.txt ‐evalue 1.0e‐4 ‐outfmt 6 ‐num_alignments 1;
blastn ‐db blast‐v4.cds.all.fa ‐query AD‐test.fa ‐out AD‐test2cdsblast.txt ‐evalue 1.0e‐4 ‐outfmt 6 ‐num_alignments 1
```

- step5: Obtain BD-AD Interactions  
```sh
perl ccs-bd-ad.pl BD‐test2cds‐blast.txt AD‐test2cds‐blast.txt >ccs‐genepairtest.txt ##ccs-bd-ad.pl was in Section0/code
``` 

- step6: Obtain unique undirected gene pairs  
```sh
perl genepair-uniq.pl ccs‐genepairtest.txt > edge-proteome-lowconf.txt ##genepair-uniq.pl was in Section0/code  
``` 
- step7: Obtain high confidence gene pairs 
> 空载实验被用来去除在BD文库中可能的自激活基因，最终的输出文件是edge-proteome-highconf.txt在Section0/data
>>Empty vector assay was used to remove possible self-activating genes only existing in BD library, and final output file was edge-proteome-highconf.txt in Section0/data.

#### Pipeline-ribo: Pipeline of calculating expression of RNAs being translated by Ribo-seq
> 清洗数据提交到了[GSA数据库](https://ngdc.cncb.ac.cn/gsa/),CRA006221。
>> Clean data was submitted to GSA database (https://ngdc.cncb.ac.cn/gsa/) with accession number CRA006221. Final output expression file (translatome.tpm) was uploaded into section0/data.
- step1: mapping to the genome  比对到基因组
```sh
module load hisat2/2.1.0; module load SAMtools/1.7;
hisat2 -x B73_v4_index --dta -p 8 -U clean_data/$i\.clean.fq.gz | samtools sort -@ 8 -o hisat+stringtie/samtools_$i\_sort.bam
samtools view -@ 8 -b -q 50 hisat+stringtie/samtools_$i\_sort.bam > hisat+stringtie/samtools_$i\_sort_uq.bam;
```
- step2: calculate gene expression  计算表达量
```sh
module load StringTie/1.3.0-foss-2016b; stringtie hisat+stringtie/samtools_$i\_sort_uq.bam -p 10 -G cds-AGPv4.40.gtf -e -A hisat+stringtie/$i\_gene_exp.tab; rm hisat+stringtie/samtools_$i\_sort.bam
```
- step3: merge all the tissues' expression to one file  合并所有组织表达量
```sh
perl mergetpm.pl > 21+21ribo.tpm ##The code refers to mergetpm.pl in the code folder, and slightly modified according to the array location.
```  
#### Pipeline-srnacluster: Pipeline of identifying and quantifing small RNA clusters by sRNA-seq
> 原始数据在[GSA数据库](https://ngdc.cncb.ac.cn/gsa/),CRA006221。最后的输出表达量数据21tis-srna.tpm.gz和注释文件在section0/data
>> Raw data was submitted to GSA database (https://ngdc.cncb.ac.cn/gsa/) with accession number CRA006221. Final output expression file (21tis-srna.tpm.gz) and annotation file (srna.gff3.gz) were uploaded into section0/data.
- step1: quality control and remove adapter  质控去接头
```sh
module load cutadapt/1.9.1;
cutadapt -q 20,20 -o out.fq.gz input.fq.gz;  cutadapt -a AGTCGGAGGCCAAGCGGTCTTAGGAA -o out-rmada.fq.gz --discard-untrimmed -e 0.05 -O 14 --no-indels out.fq.gz;
cutadapt -a "A{10}" --no-trim --untrimmed-o out-rmada-rmpolya.fq.gz -m 15 -e 0 -O 10 --no-indels --max-n 0.1 out-rmada.fq.gz
```  
  
- step2: filter sequences not in the 18-26nt  #保留18-26nt的序列
```sh
seqkit seq out.fq.gz -m 18 -M 26 -g -o trim18-26/out.fq.gz;
seqkit fq2fa trim18-26/out.fq.gz -w 0 >01trim18-26/out.fa
```
  
- step3: Compress into uniq sequence  #序列去重
```sh
perl srnauniq.pl 01trim18-26/out.fa >01trim18-26/uniq-out.fa  ##srnauniq.pl is in the code directory  
```
- step4: remove which mapping to rRNA, tRNA, snRNA, snoRNA sequences and then mapping to genome file  #去除rRNA、tRNA、snRNA、snoRNA序列并比对到基因组文件
```sh
module load Bowtie/1.2.2; 
module load SAMtools/1.7; 
bowtie -f -a -v 0 -p 1 --un 02rm-nc/rmnc-out.fa bowtie-index/zmnc-idsimple 01trim18-26/uniq-out.fa 02rm-nc/out.bwt 2>02rm-nc/out.log; 
rm 02rm-nc/out.bwt; 
bowtie -a -m 1 -f -S -v 0 -p 1 --al 03togenome/align-togenome-out.fa bowtie-1.2.2-index/Zea_mays.AGPv4.dna.toplevel 02rm-nc/rmnc-out.fa 2> 03togenome/out.log | samtools view -@ 1 -b -S -h | samtools sort -@ 1 -o 03togenome/uniq-sort-out.bam
```
- step5: merge all the bams  合并所有bam文件
```sh
module load SAMtools/1.7; samtools merge -@ 10 03mergeuniq.bam 03togenome/uniq-sort-out1.bam 03togenome/uniq-sort-out2.bam 03togenome/uniq-sort-out3.bam ...
```
- step6: bamtobed  bam转成bed文件
```sh
module load BEDTools/2.27; bedtools bamtobed -cigar -i 03mergeuniq.bam > 03mergeuniq.bed
```
- step7: Remove low coverage sites (<6x)  去除低覆盖率位点
```sh
perl srnacoverage.pl 03mergeuniq.bed >03mergeuniq-6x.bed  ##srnacoverage.pl is in the code directory  
```
- step8: merge adjacent bed to cluster  合并临接bed到cluster
```sh
bedtools merge -d 300 -i 03mergeuniq-6x.bed > 03mergeuniq-6x-gap300.bed  
```
- step9: calculate read counts in each cluster  每个簇计算read计数
```sh
bedtools intersect -a 03mergeuniq.bed -b 03mergeuniq-6x-gap300.bed -wb >cluster-tissues-read.txt;  
bedtools intersect -a 03mergeuniq.bed -b 03mergeuniq-6x.bed -wb >highcoverage.bed;  
perl srnacount.pl >cluster-readcount.txt  ##srnacount.pl is in the code directory  
```  
- step10: transfer read counts to TPM  read计数转成TPM
```sh
perl srnatoTPM.pl 03mergeuniq-6x-gap300.bed clusterinfo.txt >21+21srnacluster.tpm  ##srnatoTPM.pl is in the code directory
```
#### Pipeline-totalrna: Pipeline of calculating expression of total RNAs by mRNA-seq
> 原始数据在[GSA数据库](https://ngdc.cncb.ac.cn/gsa/),CRA006221。最后的输出表达量数据slim-transcriptome.tpm在section0/data
>> Raw data was submitted to GSA database (https://ngdc.cncb.ac.cn/gsa/) with accession number CRA006221. Final output expression file (slim-transcriptome.tpm) was uploaded into section0/data.
- step1: quality control and remove adapter  质控去接头
```sh
module load SOAPnuke/2.1.6;  
SOAPnuke filter -J -T 3 -n 0.1 -q 0.5 -l 12 -1 totalrna/raw-1.fq.gz -2 totalrna/raw-2.fq.gz  -f AAGTCGGAGGCCAAGCGGTCTTAGGAAGACAA  -r AAGTCGGATCGTAGCCATGTCGTTCTGTGAGCCAAGGAGTTG  -o 01SOAPnuke/out -C clean-out_1.fq.gz -D clean-out_2.fq.gz;    
  ```
- step2: mapping to the genome  比对到基因组
```sh
module load hisat2/2.1.0; module load SAMtools/1.7;  
hisat2 -x hisat_index/B73_v4_index --dta -p 5 --rna-strandness RF -1 01SOAPnuke/out/clean-out_1.fq.gz -2 01SOAPnuke/out/clean-out_2.fq.gz --known-splicesite-infile hisat_index/B73_splicesites.txt | samtools sort -@ 5 -o hisat+rf/samtools_out_sort.bam;   
samtools view -@ 5 -b -q 50 hisat+rf/samtools_out_sort.bam > hisat+rf/samtools_out_sort_uq.bam;  
```
- step3: calculate gene expression  计算基因表达
```sh
module load StringTie/1.3.0-foss-2016b;
stringtie hisat+rf/samtools_out_sort_uq.bam -p 5 -G Zea_mays.AGPv4.40.gtf -e -A hisat+rf/out_gene_exp.tab; rm hisat+rf/samtools_out_sort.bam  
```
- step4: merge all the tissues' expression to one file  合并所有表达量
```sh
perl mergetpm.pl > 31+31totalrna.tpm  ##mergetpm.pl is in the code directory
```
### Section1
> 六种不同类型的网络(精简的共表达网络、共翻译网络、携有ncRNA的共表达网络、互作组、精简的综合组学、携有ncRNA的综合组学)被构建
>> Six different type networks (Slim co-expression network, Co-translation network, Co-expression network with ncRNA, Interactome, Slim-IntegrativeOmics, IntegrativeOmics with ncRNAs) were constructed.  
>1. 输入文件(log-transformed expression matrixs: totalrna-log-rep1.txt, totalrna-log-rep2.txt, ribo-log-rep1.txt, ribo-log-rep2.txt, rnawithnc-exp.txt)用于共表达和共翻译网络。
>2. 输入文件(edge-proteome-highconf.txt, edge-proteome-lowconf.txt)用于蛋白互作组。
>>Input files (log-transformed expression matrixs: totalrna-log-rep1.txt, totalrna-log-rep2.txt, ribo-log-rep1.txt, ribo-log-rep2.txt, rnawithnc-exp.txt) for co-expression and co-translation network and input files (edge-proteome-highconf.txt, edge-proteome-lowconf.txt) for Interactome were uploaded into Section0/data. 
> 所有输出的edge文件存放在[NCBI](https://www.ncbi.nlm.nih.gov/geo/),用的GSE199932作为GEO种质编号
>> All final output edge files were stored into NCBI (https://www.ncbi.nlm.nih.gov/geo/) under GEO accession number GSE199932.  
> 网络特征计算：节点信息文件(Section1/data)、模块文件(Section1/data)、最短距离(GEO accession number GSE199932)
>> Network properties were calculated: node information files (uploaded into Section1/data), module files (uploaded into Section1/data), shortest distance files (in GEO with accession number GSE199932).  

#### Net1: Slim co-expression network(only containing annotated genes)  
- step1: calculate soft thresholding  
```sh
module load R/3.6.0;  
Rscript softThresholding.r totalrna-log-rep1.txt totalrna-softThresholding-rep1.pdf;  
Rscript softThresholding.r totalrna-log-rep2.txt totalrna-softThresholding-rep2.pdf   
``` 
- step2: sample clustering to detect outliers
```sh
module load R/3.6.0;  
Rscript sampletodetectoutliers.r totalrna-log-rep1.txt totalrna-Sampletodetectoutliers-rep1.pdf;  
Rscript sampletodetectoutliers.r totalrna-log-rep2.txt totalrna-Sampletodetectoutliers-rep2.pdf  
```
- step3: constrcut co-expression network by WGCNA
```sh
module load R/3.6.0;  
Rscript totalrna-edge.r totalrna-log-rep1.txt 18 totalrna-edge-rep1.txt;  
Rscript totalrna-edge.r totalrna-log-rep2.txt 12 totalrna-edge-rep2.txt  
```
- step4: obtain overlapped gene-gene pairs as final co-expression network's edges  
```sh
perl network-overlap.pl totalrna-edge-rep1.txt totalrna-edge-rep2.txt >edge-slimcoexpression.txt   
```
- step5: statistics node information
```sh
module load R/3.6.0;  
Rscript NetInfo.r edge-slimcoexpression.txt nodeinfo-slimcoexpression.txt sd-slimcoexpression.txt transitivity-slimcoexpression.txt
```
- step6: divide modules
```sh
module load MCL/14-137; mcl edge-slimcoexpression-abc.txt --abc -o module-slimcoexpression.txt
```

#### Net2: Co-translation network  
- step1: calculate soft thresholding  
```sh
module load R/3.6.0;  
Rscript softThresholding.r ribo-log-rep1.txt ribo-softThresholding-rep1.pdf; 
Rscript softThresholding.r ribo-log-rep2.txt ribo-softThresholding-rep2.pdf  
```
- step2: sample clustering to detect outliers
```sh
module load R/3.6.0;  
Rscript sampletodetectoutliers.r ribo-log-rep1.txt ribo-Sampletodetectoutliers-rep1.pdf;  
Rscript sampletodetectoutliers.r ribo-log-rep2.txt ribo-Sampletodetectoutliers-rep2.pdf  
```
- step3: constrcut co-translation network by WGCNA
```sh
module load R/3.6.0;  
Rscript totalrna-edge.r ribo-log-rep1.txt 9 ribo-edge-rep1.txt;  
Rscript totalrna-edge.r ribo-log-rep2.txt 21 ribo-edge-rep2.txt 
```
- step4: select overlapped gene pairs as final co-expression network's edges  
```sh
perl network-overlap.pl ribo-edge-rep1.txt ribo-edge-rep2.txt >edge-cotranslation.txt   
```
- step5: statistics node information
```sh
module load R/3.6.0;  
Rscript NetInfo.r edge-cotranslation.txt nodeinfo-cotranslation.txt sd-cotranslation.txt transitivity-cotranslation.txt 
```
- step6: divide modules
```sh
module load MCL/14-137; mcl edge-cotranslation-abc.txt --abc -o module-cotranslation.txt
```
#### Net3: Co-expression network with ncRNA(not only including mRNA, but also lncRNA, small RNA cluster, circRNA, fusionRNA) 
- step1: construct network by one step
```sh
module load R/3.6.0; Rscript wgcna-edge.r rnawithnc-exp.txt edge-coexpressionwithncrna.txt  
```
- step2: statistics node information
```sh
module load R/3.6.0; Rscript NetInfo.r edge-coexpressionwithncrna.txt nodeinfo-coexpressionwithncrna.txt sd-coexpressionwithncrna.txt transitivity-coexpressionwithncrna.txt
```
- step3: divide modules
```sh
module load MCL/14-137; mcl edge-coexpressionwithncrna-abc.txt --abc -o module-coexpressionwithncrna.txt
```
#### Net4: Interactome(high confidence and low confidence)  
- step1: statistics node information
```sh
module load R/3.6.0; Rscript NetInfo.r edge-proteome-highconf.txt nodeinfo-proteome-highconf.txt sd-PPIs-highconf.txt transitivity-PPIs-highconf.txt  
module load R/3.6.0; Rscript NetInfo.r edge-proteome-lowconf.txt nodeinfo-proteome-lowconf.txt sd-PPIs-lowconf.txt transitivity-PPIs-lowconf.txt  
```
- step2: divide modules
```sh
module load MCL/14-137; mcl edge-proteome-highconf-abc.txt --abc -o module-proteome-highconf.txt  
module load MCL/14-137; mcl edge-proteome-lowconf-abc.txt --abc -o module-proteome-lowconf.txt  
```
#### Net5: Slim-IntegrativeOmics(high confidence and low confidence)  
- step1: statistics node information  
>整合每个组学所有的基因对，ChIA-pet网络、精简的共表达网络、共翻译网络和互作组，最终的输出结果文件(edge-slimio-highconf.txt and edge-slimio-lowconf.txt)上传到GEO的种质号GSE199932。
>>Integrates all gene-gene pairs of each omics, ChIA-PET network, slim co-expression network, co-translation network and interactome (high and low confidences). Final output files (edge-slimio-highconf.txt and edge-slimio-lowconf.txt) were uploaded into GEO with accession number GSE199932.
- step2: statistics node information
```sh
module load R/3.6.0; Rscript NetInfo.r edge-slimio-highconf.txt nodeinfo-slimio-highconf.txt sd-slimio-highconf.txt transitivity-slimio-highconf.txt  
module load R/3.6.0; Rscript NetInfo.r edge-slimio-lowconf.txt nodeinfo-slimio-lowconf.txt sd-slimio-lowconf.txt transitivity-slimio-lowconf.txt  
```
- step3: divide modules
```sh
module load MCL/14-137; mcl edge-slimio-highconf-abc.txt --abc -o module-slimio-highconf.txt  
module load MCL/14-137; mcl edge-slimio-lowconf-abc.txt --abc -o module-slimio-lowconf.txt  
```
#### Net6: IntegrativeOmics with ncRNAs(high confidence and low confidence)  
- step1: statistics node information  
>整合每个组学所有的基因对，ChIA-pet网络、携有非编码RNA的共表达网络、共翻译网络和互作组，最终的输出结果文件(edge-iowithncrna-highconf.txt and edge-iowithncrna-lowconf.txt)上传到GEO的种质号GSE199932。
>>Integrates all gene-gene pairs of each omics, ChIA-PET network, co-expression network with ncRNA, co-translation network and interactome (high and low confidences).Final output files (edge-iowithncrna-highconf.txt and edge-iowithncrna-lowconf.txt) were uploaded into GEO with accession number GSE199932.
- step2: statistics node information 统计节点信息
```sh
module load R/3.6.0;  
Rscript NetInfo.r edge-iowithncrna-highconf.txt nodeinfo-iowithncrna-highconf.txt sd-iowithncrna-highconf.txt transitivity-iowithncrna-highconf.txt  
Rscript NetInfo.r edge-iowithncrna-lowconf.txt nodeinfo-iowithncrna-lowconf.txt sd-iowithncrna-lowconf.txt transitivity-iowithncrna-lowconf.txt 
```
- step3: divide modules 分模块
```sh
module load MCL/14-137; mcl edge-iowithncrna-highconf-abc.txt --abc -o module-iowithncrna-highconf.txt  
module load MCL/14-137; mcl edge-iowithncrna-lowconf-abc.txt --abc -o module-iowithncrna-lowconf.txt  
```

### Section2
>以网络的角度进行分类和评估复制基因的功能分化
>> This section is about clustering and assessing functional divergence of duplicated genes from network perspective.  
>> analysis.md: Data Analysis and visual code for Section2;  
>>a The code and data involved in analysis.md were stored in Section2/code, Section2/duplicategenes and Section2/data.
- Statistics for Fig. 2b
```sh
perl module.pl module-slimcoexpression.txt module-cotranslation.txt module-proteome-highconf.txt
```  
>The code module.pl was in Section2/code.  
>Intermediate files of this code were five duplicate genes (Zma.dispersed.txt, Zma.proximal.txt, Zma.tandem.txt, Zma.transposed.txt and Zma.wgd.txt) in Section2/duplicategenes and a random unparalogous pair file (para_random.txt) in Section2/data.  
>Three input module files were in Section1/data.
- Statistics for Fig. 2c
```sh
perl duplicate-SD.pl sd-slimcoexpression.txt sd-cotranslation.txt sd-PPIs-highconf.txt
```  
> Three input shortest distance files were in GEO with accession number GSE199932.
- Statistics for Fig. 2d-2g
```sh
for i in $(cat id.txt);do perl genecommon.pl edge-slimcoexpression.txt nodeinfo-slimcoexpression.txt Zma.$i.txt >trans-$i.txt;  done  
for i in $(cat id.txt);do perl genecommon.pl edge-cotranslation.txt nodeinfo-cotranslation.txt Zma.$i.txt >ribo-$i.txt; done   
for i in $(cat id.txt);do perl genecommon.pl edge-proteome-highconf.txt nodeinfo-proteome-highconf.txt Zma.$i.txt >ppi-$i.txt; done  
perl dabsab.pl trans >trans-type.txt  
perl dabsab.pl ribo >ribo-type.txt  
perl dabsab.pl ppi >ppi-type.txt 
```  
>Input edge files were in GEO with accession number GSE199932; input node information files were in Section1/data; input id.txt was in Section2/data; code genecommon.pl and dabsab.pl were in Section2/code.  
- Visual code for Fig. 2g and Extended Data Fig. 13
```R
library(ggalluvial)
library(ggplot2)
a <- read.table("wgd-change.txt",header=T,sep="\t") ##wgd-change.txt in Section2/data
ggplot(a,aes(axis1=trans,axis2=ribo,axis3=ppi))+
  geom_alluvium(aes(fill = trans))+
  geom_stratum()+
  geom_text(stat = "stratum",aes(label = after_stat(stratum)))+
  scale_x_continuous(breaks = 1:3, labels = c("Co-expression", "Co-translation", "Interactome"))+
  theme_bw()+theme(legend.position="none")
```
### Section3
> 两个玉米亚基因组间的不对称调控分化
>> This section is about asymmetric regulatory divergence between the two maize subgenomes.  
>> analysis.md: Data Analysis and visual code for Section3; The code and data involved in analysis.md were stored in Section3/code and Section3/data.

#### Statistics and visual code for Fig. 3b and Extended Data Fig. 14
- step1:  
```sh
awk -F '\t' '{if ($2==1) print $0}' 00sbicolor_maize-omics.txt |cut -d " " -f 8-10 >1.txt` ##Obtain information of chromosome1 
```
- step2:  
```sh
perl slindingwindow.pl 10 100 >chr1.txt` ##slindingwindow.pl Section3/code and 00sbicolor_maize-omics Section3/data
```  
step3: 
```R
##figure by R, ##chr1.txt was in data folder
a <- read.table("chr1.txt",header = T,sep = "\t")
a$omics <- factor(a$omics, levels = c("Co-expression","Co-translation","Interactome"))
ggplot(a,aes(x=id,y=per,color=type))+
	geom_point(size=0.5)+
	geom_smooth(formula = y ~ x,method="loess",size=1.5)+
	scale_color_manual(values=c("#E7B800", "#2E9FDF"))+
	labs(x="Gene order along Sorghum bicolor chr1",y="Proportion of dominant subgenome in a bin")+
	facet_grid(.~omics,as.table=F)+
	theme_bw()+theme(legend.position = "none")
```

### Section4
> 预测基因功能，重构调控网络和计算网络分类显著性
>>This section is about prediction gene function, reconstruction regulating networks and calculation network cluster's significance.  
>>analysis.md: Data Analysis and visual code for Section4;
>>The code and data involved in analysis.md were stored in Section4/code and Section4/data.

#### 1. Code for Fig. 4f and Extended Data Fig. 18
- step1: construct a direct-connected network
```sh
perl net-direct.pl 62kernel.txt out.csv
```  
##net-direct.pl was in Section4/code; 62kernel.txt containing 62 kernel genes was in Section4/data; and edge-slimio-lowconf.txt in this code was in GEO with accession number GSE199932.
- step2:calculate the significant
```sh
perl net-pavlue.pl 62 1000 62kernel.txt
```   
##net-pavlue.pl was in Section4/code. 62 is the number of kernel genes and 1000 is the number of random iterations.

#### 2. Code for Extended Data Fig. 17a
```sh
for ((i=1;i<=1000;i++));do perl 45+17test.pl 62kernel.txt ;done
```

### Section5
> 通过机器学习预测开花时间基因
>>This section is about predicting genes of flowering time by machine learning.  
>>The input files train.xlsx and test.xlsx involved in the following analysis were uploaded into Section5/data.
#### 1. Computation of node similarity in a network
```sh
perl similarity.pl edge-proteome-highconf.txt nodeinfo-proteome-highconf.txt traingene-34pos+34neg.txt 5 8 >out.txt
```
>similarity.pl was in Section5/code;  
>edge-proteome-highconf.txt was in Section0/data;  
>nodeinfo-proteome-highconf.txt was in Section1/data;  
>traingene-34pos+34neg.txt was in Section5/data;  
>5 and 8 means genes in 5-8 rows of nodeinfo-proteome-highconf.txt.   
>This code was used to calculate similarity with 68 train genes in the network, including:  
> 1. Common Neighbours Similarity (CN);  
> 2. Resource Allocation Similarity (RA);  
> 3. Total Neighbours Similarity (TN);  
> 4. Preferential Attachment (PA);  
> 5. Hub Promoted Similarity (HP);  
> 6. Hub Depressed Similarity (HD);  
> 7. Sorensen-Dice Similarity (sorensen). 

#### 2. Model evaluation (AUC and F1 values) of five classical machine learning algorithms 
> LogisticRegression, BaggingClassifier, XGBClassifier, SVM and Neural Network for Interactome. 
>Trainning_results_for_sd_feature.py was in Section5/code.

#### 3. Model evaluation (AUC and F1 values) of five classical machine learning algorithms 
> (LogisticRegression, BaggingClassifier, XGBClassifier, SVM and Neural Network) for Intergrative omics which containg expression of transcriptome and translatome, nodeinformation and shortest distance attributes of Interactome.
>Training_results_for_different_features.py was in Section5/code.

#### 4. Result prediction and feature importance calculation
>predresult.py was in Section5/code.

#### 5. Code for ROC-AUC figure and PR-curve
>Draw_picture.py was in Section5/code.
