---
title: 转录组分析
subtitle:
date: 2023-07-22T12:32:20+08:00
draft: false
type: posts
author:
  name:
  link:
  email:
  avatar:
description:
keywords:
license:
comment: false
weight: 0
tags:
  - 转录调控
categories:
  - 转录调控
hiddenFromHomePage: false
hiddenFromSearch: false
summary:
resources:
  - name: featured-image
    src: featured-image.jpg
  - name: featured-image-preview
    src: featured-image-preview.jpg
toc: true
math: false
lightgallery: false
password:
message:
repost:
  enable: true
  url:

# See details front matter: https://fixit.lruihao.cn/documentation/content-management/introduction/#front-matter
---
# Multiple RNA seq analysis
## Total RNA Pipeline 总RNA流程

{{< admonition type=abstract open=false title="Total RNA pipeline" >}}

```sh
hisat2-build -p 20 ${i}.fa  #参考基因组建立索引
vdb-validate ${i}.sra #下载SRA完整性的检验
fastq-dump --split-3 --gzip --defline-seq '@\$sn_\$ac'  ${i}.sra #将SRA文件拆分为双端reads
fastp  -w 1 -i ${i}_1.fastq.gz -I ${i}_2.fastq.gz -o ${i}_1.QC.fastq.gz -O ${i}_2.QC.fastq.gz #fastq文件进行质控
hisat2 -p 1 -x ${hisat_index}SL5.0 --dta -1 ${i}_1.QC.fastq.gz -2 ${i}_2.QC.fastq.gz -S ${i}.hisat2.sam --summary-file ${i}.hisat2.summary  --new-summary #对测序reads质控文件进行比对
samtools sort -@ 8 -o ${i}.hisat2.bam ${i}.hisat2.sam #sam文件排序并转bam
stringtie -e -G ${gff} -l ${i} -o ${i}.transcripts.gtf -b ${i}.ballgown -A ${i}_abundances.tsv -p 8 ${i}.hisat2.bam #对转录本进行定量 

```
- hisat2参数解析
```txt
--dta reports alignments tailored for transcript assemblers 生成用于比对组装的比对详情
--rna-strandness <string>
Specify strand-specific information: the default is unstranded.

For single-end reads, use F or R.
‘F’ means a read corresponds to a transcript.
‘R’ means a read corresponds to the reverse complemented counterpart of a transcript.

For paired-end reads, use either FR or RF.
With this option being used, every read alignment will have an XS attribute tag:
’+’ means a read belongs to a transcript on ‘+’ strand of genome.
‘-‘ means a read belongs to a transcript on ‘-‘ strand of genome.
```

- stringtie转录本组装参数解析
```txt
-e only estimate the abundance of given reference transcripts (requires -G) 限制reads比对的处理，仅估计和输出与用-G选项给出的参考转录本匹配的组装转录本。使用该选项，则会跳过处理与参考转录本不匹配的组装转录本，这将大大的提升了处理速度。
-G <guide_gff>   reference annotation to include in the merging (GTF/GFF3) 参考注释基因组文件(GTF/GFF3)
-l name prefix for output transcripts (default: STRG) 输出转录本的名称前缀
-o output path/file name for the assembled transcripts GTF (default: stdout) 指定输出组装的GTF文件的路径和名称
-b enable output of Ballgown table files but these files will be created under the directory path given as <dir_path> 输出ballgown格式的文件
-A gene abundance estimation output file 输出结果中的gene丰度信息
-p number of threads (CPUs) to use (default: 1) 线程数
-j minimum junction coverage (default: 1) ?
```
- stringtie合并转录本参数解析
```sh
stringtie --merge -G ITAG3.2_gene_models.gff -m 100 -c 0.1 -f 0.05 -o merged.gtf gtf.list 
```
```txt
-G 如上
-m minimum input transcript length to include in the merge (default: 50) 合并文件中，指定允许最小输入转录本的长度 (默认值: 50)
-c minimum input transcript coverage to include in the merge (default: 0)  合并文件中，指定允许最低输入转录本的覆盖度(默认值: 0)
-f minimum isoform fraction (default: 0.01)
-l <label> name prefix for output transcripts (default: MSTRG)
```
- 组装转录本再次定量
```sh
stringtie -e -G merged.gtf -o ${sample}_exp.gtf -A ${sample}_gene.exp ${sample}_uniq.header.bam
```
{{< /admonition >}}

## lncRNA Pipeline 长非编码流程
### Multi-omics reveals the genetic regulation and function of long non-coding RNA in tomato
To identify the lncRNAs, the short transcripts and transcripts with coding potential were filtered out through the following steps: 
1. Transcripts with lengths less than 200 bp were filtered out.
2. Transcripts with an open reading frame (ORF) length of more than 100 aa (predicted by ORF-finder) were filtered out.
3. Transcripts aligned to the Pfam or Swiss-Prot databases with a p-value < 0.001 (Kiefer et al., 2009) were filtered out.
4. Transcripts with a coding potential >0.5, determined  by the Coding Potential Calculator (CPC) (Kang et al., 2017; Kong et al., 2007), were filtered out. 
5. The strand information of the lncRNAs was predicted by StringTie based on the genomic splice sites in the case of spliced alignments (introns in the alignment), which was encoded by the aligner into the XS tag. 
6. For single-exon transcripts lacking strand information, both strand transcripts are predicted to have no protein-coding potential. 
7. This is a common approach used in lncRNA studies (Ma et al., 2021; Wang et al., 2023; Wei et al., 2022; Zhang et al., 2022b). 

- [x] 过滤: 转录本小于200bp 
- [ ] 过滤: ORF大于100aa
- [ ] 过滤: 转录本比对到Pfam或SwissProt数据库p值小于0.001的
- [ ] 过滤：编码可能性大于0.5 通过Coding Potential Calculator计算
- [ ] 通过stringtie预测链信息
- [ ] 缺链信息的单外显子转录本，两条转录本被预测没有编码潜能
### Maize Li lin

- The improved version of LncRNA_Finder<sup>58</sup> was used to identify lncRNAs from total RNA.
- Transcripts longer than >300 nucleotides were removed because >95% of the protein-coding genes encode proteins with >100 amino acids<sup>59</sup>.
- The remaining transcripts were used as an input for CPC2 <sup>60</sup>, which rapidly and accurately assesses the coding ability of RNAs to discriminate and remove potential protein-coding transcripts.
- To remove housekeeping RNAs, including tRNAs (http://gtrnadb.ucsc.edu/), rRNAs (http://maize.jcvi.org/), snRNAs and snoRNAs (http://noncode.org/); the remaining transcripts were compared to the listed housekeeping RNA databases.
- The remaining transcripts were considered to be lncRNAs.
- StringTie<sup>57</sup> was used to quantify lncRNA expression levels with the parameters -G -e --A.

{{< admonition type=abstract open=false title="lncRNA_Finder" >}}  
- lncRNA_Finder.pl源码
> 1. [lncRNA_Finder](https://github.com/caulilin/lncRNA_Finder/blob/master/LncRNA_Finder.pl) #所需文件 housekeeping; sRNA; 转录本序列; 蛋白序列；
> 2. [LncRNA_Finder2.0](https://github.com/hanlinqian/LncRNA_Finder2.0/blob/master/LncRNA_Finder2.0.pl) #无需sRNA输入文件

- lncRNA_Finder.pl依赖软件: ncbi_blast; bowtie; cpc

- lncRNA_Finder.pl 参数解析
``` txt
   -i \<transcript.fasta\> 转录本fasta文件

   -p \<protein.fasta\> 蛋白fasta文件

   -k \<housekeeping.fasta\> 管家基因fasta文件

   -s \<smallRNA.fasta\> 小RNAfasta文件

   -o \<output prefix\> 输出前缀

   -t <int> number of thread for the computation || default=4 线程数

   -r <int> minimum lncRNA length || default=200 最小lncRNA长度默认200

   -f <int> maximum potential ORF length of lncRNAs || default=100 lncRNA蛋白ORF的最大长度

   -m <int> number of mismatch in the alignment with smallRNA || default=0 与小RNA错误匹配的数量

   -e E-value of the alignment against protein database || default=1.0e-3 蛋白数据库比对e-value值
```
{{< /admonition >}}
